<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Individual Based Genetic Distance for SNP Data</title>

<script src="libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.5/css/spacelab.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="libs/navigation-1.1/tabsets.js"></script>
<!---

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
  padding-right: 50px;
  padding-left: -50px;
}
@media (max-width: 979px) {
  body {
    padding-top: 0;
  }
}

 offset scroll position for anchor links (for fixed navbar)  
@media (min-width: 980px) {
  .section h2 {
    padding-top: 52px;
    margin-top: -52px;
  }
  .section h3 {
    padding-top: 52px;
    margin-top: -52px;
  }
}


/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

</style>
-->



<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}
/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}
/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}
</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #f8f8f8; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
pre, code { background-color: #f8f8f8; }
code > span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code > span.dt { color: #204a87; } /* DataType */
code > span.dv { color: #0000cf; } /* DecVal */
code > span.bn { color: #0000cf; } /* BaseN */
code > span.fl { color: #0000cf; } /* Float */
code > span.ch { color: #4e9a06; } /* Char */
code > span.st { color: #4e9a06; } /* String */
code > span.co { color: #8f5902; font-style: italic; } /* Comment */
code > span.ot { color: #8f5902; } /* Other */
code > span.al { color: #ef2929; } /* Alert */
code > span.fu { color: #000000; } /* Function */
code > span.er { color: #a40000; font-weight: bold; } /* Error */
code > span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #000000; } /* Constant */
code > span.sc { color: #000000; } /* SpecialChar */
code > span.vs { color: #4e9a06; } /* VerbatimString */
code > span.ss { color: #4e9a06; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #000000; } /* Variable */
code > span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code > span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code > span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code > span.ex { } /* Extension */
code > span.at { color: #c4a000; } /* Attribute */
code > span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code > span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <!-- HOME -->
      <a class="navbar-brand" href="index.html"><b>Population Genetics in R</b></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <!-- Table of contents -->
        <li class="dropdown">
          <a href="chapters" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Users <span class="caret"></span></a>
          <ol class="dropdown-menu">
              <li><a href="GETSTARTED.html">Getting Started</a></li>
              <li class="divider"></li>
              <li class="dropdown-header">For microsatellite data:</li>
              <li><a href="startMicrosatellite.html">Basic statistics</a></li>
              <li><a href="2015-12-15-microsatellite-differentiation.html">Population differentiation</a></li>
              <li><a href="LMM-Genetic-Diversity.html">Comparing genetic diversity using mixed models</a></li>
              <li class="divider"></li>
              <li class="dropdown-header">For SNP data:</li>
              <!-- <li class="divider"></li> -->
              <li><a href="StartSNP.html">Basic statistics</a></li>
              <li><a href="DifferentiationSNP.html">Population differentiation</a></li>
              <li><a href="2015-05-18-Dist-SNP.html">Genetic distance (individual-based)</a></li>
              <li><a href="2016-01-26-SNP-selection.html">Signals of selection</a></li>
			  <li><a href="2016-12-13_MEM_outlier.html">Spatial outlier detection</a></li>
			  <li><a href="2018-03-27_RDA_GEA.html">Detecting multilocus adaptation using redundancy analysis</a></li>
              <!-- <li><a href="SNP.html">SNPs</a></li> -->
              <li class="divider"></li>
              <li class="dropdown-header">For sequence data:</li>
              <!-- <li class="divider"></li>   -->
              <li><a href="PopDiffSequenceData.html">Population differentiation</a></li>
          </ol>
        </li>
        <!-- Appendices dropdown -->
        <li class="dropdown">
          <a href="appendices" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Package Developers <span class="caret"></span></a>
          <ul class = "dropdown-menu" role = "menu">
           <li><a href = "PACKAGES.html">Population genetic packages in R</a></li>  
           <li><a href = "DATAFORMATS.html">Population genetic data types in R</a></li>
           <li><a href = "PACKAGE_TUTORIALS.html">Tutorials for developing R packages</a></li>
           <li class="divider"></li>  
           <li class="dropdown-header">Development Resources (external):</li>
           <li><a href = "http://r-pkgs.had.co.nz/">Hadley Wickham's guide for creating R packages</a></li>
           <li><a href = "https://github.com/jtleek/rpackages">Jeff Leek's R package development</a></li>
           <li><a href = "http://adv-r.had.co.nz/S4.html">S4 Classes in detail</a></li>
           <li><a href = "http://adv-r.had.co.nz/S3.html">S3 Classes in detail</a></li>
           <li><a href = "http://cran.r-project.org/doc/contrib/Genolini-S4tutorialV0-5en.pdf">A not so short introduction to S4</a></li>
          </ul>
        </li>
        <!-- Author button -->
        <li class="dropdown">
          <a href="about" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Contribute! <span class="caret"></span></a>
          <ul class = "dropdown-menu" role = "menu">
           <li><a href = "CONTRIBUTING.html">Contribution guidelines</a></li>  
           <li><a href = "CONTRIBUTING_WITH_GIT2R.html">Contributing with git2r</a></li> 
           <li class="divider"></li>  
           <li><a href="https://github.com/NESCent/popgenInfo">Source repository</a></li>  
		  </ul>
        </li>
        <li class = "button">
          <a href="LinksToResources.html">Useful Links</a>
        </li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<!-- <div class="container">
<div class="row">
<div class="span10 offset1"> -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Individual Based Genetic Distance for SNP Data</h1>

</div>

<div id="TOC">
<ul>
<li><a href="#introduction">Introduction</a></li>
<li><a href="#assumptions">Assumptions</a></li>
<li><a href="#data">Data</a></li>
<li><a href="#resourcespackages-required">Resources/Packages required</a></li>
<li><a href="#analysis">Analysis</a><ul>
<li><a href="#section-1-convert-the-data">Section 1: Convert the data</a></li>
<li><a href="#section-2.-individual-genetic-distance-euclidean-distance-dist-adegenet">Section 2. Individual genetic distance: euclidean distance (<code>dist</code> {adegenet})</a></li>
<li><a href="#section-3.-individual-genetic-distance-number-of-loci-for-which-individuals-differ-dist.gene-ape">Section 3. Individual genetic distance: number of loci for which individuals differ (<code>dist.gene</code> {ape})</a></li>
<li><a href="#section-4-number-of-allelic-differences-between-two-individuals-diss.dist-poppr">Section 4: number of allelic differences between two individuals (<code>diss.dist</code> {poppr})</a></li>
<li><a href="#section-5-conclusions-drawn-from-the-analysis">Section 5: Conclusions drawn from the analysis</a></li>
</ul></li>
<li><a href="#contributors">Contributors</a></li>
<li><a href="#session-information">Session Information</a></li>
</ul>
</div>

<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>In this vignette, we will estimate individual genetic distances from SNP data. It is useful when you have individual genotype data and you don’t know the populations.</p>
</div>
<div id="assumptions" class="section level1">
<h1>Assumptions</h1>
<p>We will use non-evolutionary genetic distances, i.e. not based on Hardy-Weinberg assumptions: free from assumptions.</p>
</div>
<div id="data" class="section level1">
<h1>Data</h1>
<p>The dataset used for those analysis concerns the plant: lodgepole pine (<em>Pinus contorta</em>, <em>Pinaceae</em>). You can have more information on this data set and the species on the web site of A. Eckert: (<a href="http://eckertdata.blogspot.fr/" class="uri">http://eckertdata.blogspot.fr/</a>). But here the dataset is used as a test dataset with no idea of interpreting the results in a biological way. We will work on a subset of the dataset to make the calculations faster.</p>
<p>The data are stored in a text file (genotype = AA). We will import the dataset in R as a data frame, and then convert the SNP data file into <a href="http://www.inside-r.org/packages/cran/adegenet/docs/.valid.genind">genind</a> objects.</p>
<p>The dataset <a href="https://github.com/NESCent/popgenInfo/tree/master/data/Master_Pinus_data_genotype.txt">“Master_Pinus_data_genotype.txt” can be downloaded here</a>.</p>
<p>The text file is a matrix of (550 rows x 3086 columns). It contains 4 extra columns: first column is the label of the individuals, the three other are description of the region, all the other columns are for the genotypes as (AA or AT…).</p>
<p>When you import the data into R, the data file needs to be in your working directory, or adjust the path in the <code>read.table()</code> invocation below accordingly.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">Mydata &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;Master_Pinus_data_genotype.txt&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">check.names =</span> <span class="ot">FALSE</span>)   
<span class="kw">dim</span>(Mydata) </code></pre></div>
<pre><code>## [1]  550 3086</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ind &lt;-<span class="st"> </span><span class="kw">as.character</span>(Mydata<span class="op">$</span>tree_id) <span class="co"># individual labels </span>
population &lt;-<span class="st"> </span><span class="kw">as.character</span>(Mydata<span class="op">$</span>state) <span class="co"># population labels</span>
county &lt;-<span class="st"> </span>Mydata<span class="op">$</span>county 
<span class="kw">dim</span>(Mydata) <span class="co"># 550 individuals x 3082 SNPs</span></code></pre></div>
<pre><code>## [1]  550 3086</code></pre>
</div>
<div id="resourcespackages-required" class="section level1">
<h1>Resources/Packages required</h1>
<p>Loading the required packages:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;poppr&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;pegas&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;ape&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;adegenet&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;ade4&quot;</span>)</code></pre></div>
</div>
<div id="analysis" class="section level1">
<h1>Analysis</h1>
<div id="section-1-convert-the-data" class="section level2">
<h2>Section 1: Convert the data</h2>
<p>To work with the data, we need to convert the R object returned by <code>read.table()</code> to a <code>genind</code> object. To achieve this, we create a matrix with only genotypes, and keep only a subset of the first 100 SNP loci (to make calculations faster). The result can then be converted to a <code>genind</code> object (for the package adegenet). The <code>genind</code> object can then easily be converted into <code>loci</code> objects (package pegas) (i.e. Mydata2)</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">locus &lt;-<span class="st"> </span>Mydata[, <span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">4</span>, <span class="dv">105</span><span class="op">:</span><span class="kw">ncol</span>(Mydata))] 
Mydata1 &lt;-<span class="st"> </span><span class="kw">df2genind</span>(locus, <span class="dt">ploidy =</span> <span class="dv">2</span>, <span class="dt">ind.names =</span> ind, <span class="dt">pop =</span> population, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>)
Mydata1</code></pre></div>
<pre><code>## /// GENIND OBJECT /////////
## 
##  // 550 individuals; 100 loci; 200 alleles; size: 519.8 Kb
## 
##  // Basic content
##    @tab:  550 x 200 matrix of allele counts
##    @loc.n.all: number of alleles per locus (range: 2-2)
##    @loc.fac: locus factor for the 200 columns of @tab
##    @all.names: list of allele names for each locus
##    @ploidy: ploidy of each individual  (range: 2-2)
##    @type:  codom
##    @call: df2genind(X = locus, sep = &quot;&quot;, ind.names = ind, pop = population, 
##     ploidy = 2)
## 
##  // Optional content
##    @pop: population of each individual (group size range: 4-177)</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">Mydata2 &lt;-<span class="kw">genind2loci</span>(Mydata1)</code></pre></div>
</div>
<div id="section-2.-individual-genetic-distance-euclidean-distance-dist-adegenet" class="section level2">
<h2>Section 2. Individual genetic distance: euclidean distance (<code>dist</code> {adegenet})</h2>
<p>The unit of the observation is the individuals.</p>
<ul>
<li>550 genotypes</li>
<li>100 binary SNPs</li>
<li>Ploidy : 2</li>
</ul>
<p>The analysis is applied on allele frequency within individuals as represented in the <code>genind</code> object. We can use the function <code>dist()</code> from adegenet which provides different options. We will use the euclidean distance among vector of allele frequencies.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">distgenEUCL &lt;-<span class="st"> </span><span class="kw">dist</span>(Mydata1, <span class="dt">method =</span> <span class="st">&quot;euclidean&quot;</span>, <span class="dt">diag =</span> <span class="ot">FALSE</span>, <span class="dt">upper =</span> <span class="ot">FALSE</span>, <span class="dt">p =</span> <span class="dv">2</span>)
<span class="kw">hist</span>(distgenEUCL)</code></pre></div>
<p><img src="2015-05-18-Dist-SNP_files/figure-html/Individual_dist_adegenet-1.png" width="960" /></p>
</div>
<div id="section-3.-individual-genetic-distance-number-of-loci-for-which-individuals-differ-dist.gene-ape" class="section level2">
<h2>Section 3. Individual genetic distance: number of loci for which individuals differ (<code>dist.gene</code> {ape})</h2>
<p>The option <code>pairwise.deletion = FALSE</code> in the command <code>dist.gene()</code> removes all loci with one missing values : you an see on the histogram that we get a maximum distance of 3 loci out of 100.</p>
<p>We can see that we get 98 loci with at least one sample missing. Then using the option <code>pairwise.deletion = TRUE</code> in the command <code>dist.gene()</code> allows you to keep loci with one missing value.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">distgenDIFF &lt;-<span class="st"> </span><span class="kw">dist.gene</span>(Mydata2, <span class="dt">method=</span><span class="st">&quot;pairwise&quot;</span>, <span class="dt">pairwise.deletion =</span> <span class="ot">FALSE</span>, <span class="dt">variance =</span> <span class="ot">FALSE</span>)
<span class="kw">hist</span>(distgenDIFF)</code></pre></div>
<p><img src="2015-05-18-Dist-SNP_files/figure-html/Individual_dist_ape-1.png" width="960" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Get percent missing data per population</span>
missing_data &lt;-<span class="st"> </span><span class="kw">info_table</span>(Mydata1, <span class="dt">type =</span> <span class="st">&quot;missing&quot;</span>)
<span class="kw">sum</span>(missing_data[<span class="st">&quot;Total&quot;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">100</span>] <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>)</code></pre></div>
<pre><code>## [1] 98</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">barplot</span>(missing_data[<span class="st">&quot;Total&quot;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">100</span>], <span class="dt">xlab =</span> <span class="st">&quot;Locus&quot;</span>, <span class="dt">ylab =</span> <span class="st">&quot;Percent Missing&quot;</span>)</code></pre></div>
<p><img src="2015-05-18-Dist-SNP_files/figure-html/Individual_dist_ape-2.png" width="960" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">distgenDIFF &lt;-<span class="st"> </span><span class="kw">dist.gene</span>(Mydata2, <span class="dt">method=</span><span class="st">&quot;pairwise&quot;</span>, <span class="dt">pairwise.deletion =</span> <span class="ot">TRUE</span>, <span class="dt">variance =</span> <span class="ot">FALSE</span>)
<span class="kw">hist</span>(distgenDIFF)</code></pre></div>
<p><img src="2015-05-18-Dist-SNP_files/figure-html/Individual_dist_ape-3.png" width="960" /></p>
</div>
<div id="section-4-number-of-allelic-differences-between-two-individuals-diss.dist-poppr" class="section level2">
<h2>Section 4: number of allelic differences between two individuals (<code>diss.dist</code> {poppr})</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">distgenDISS &lt;-<span class="st"> </span><span class="kw">diss.dist</span>(Mydata1, <span class="dt">percent =</span> <span class="ot">FALSE</span>, <span class="dt">mat =</span> <span class="ot">FALSE</span>)
<span class="kw">hist</span>(distgenDISS)</code></pre></div>
<p><img src="2015-05-18-Dist-SNP_files/figure-html/Individual_dist_poppr-1.png" width="960" /></p>
</div>
<div id="section-5-conclusions-drawn-from-the-analysis" class="section level2">
<h2>Section 5: Conclusions drawn from the analysis</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">boxplot</span>(distgenEUCL, distgenDIFF, distgenDISS)</code></pre></div>
<p><img src="2015-05-18-Dist-SNP_files/figure-html/Compare_dist-1.png" width="960" /></p>
<p>The number of allelic differences between two individuals is a different measure from euclidean distance or number of locus differences between two individuals.</p>
<p>Conclusion</p>
<div id="what-did-we-learn-today" class="section level3">
<h3>What did we learn today?</h3>
<p>In this vignette, we explore different measures of individual genetic distances. It is important to investigate the different options of each command. Missing data can be handled in different ways.</p>
</div>
<div id="what-is-next" class="section level3">
<h3>What is next?</h3>
<p>These individuals genetic distances can then be used in analyses such as Mantel tests to test for isolation by distance, more complex analyses in landscape genetics to test for resistance by distance, cluster analysis, or spatial networks.</p>
</div>
</div>
</div>
<div id="contributors" class="section level1">
<h1>Contributors</h1>
<ul>
<li>Stéphanie Manel</li>
<li>Zhian Kamvar (minor edits)</li>
</ul>
</div>
<div id="session-information" class="section level1">
<h1>Session Information</h1>
<p>This shows us useful information for reproducibility. Of particular importance are the versions of R and the packages used to create this workflow. It is considered good practice to record this information with every analysis.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">width =</span> <span class="dv">100</span>)
devtools<span class="op">::</span><span class="kw">session_info</span>()</code></pre></div>
<pre><code>## Session info --------------------------------------------------------------------------------------</code></pre>
<pre><code>##  setting  value                       
##  version  R version 3.4.3 (2017-11-30)
##  system   x86_64, linux-gnu           
##  ui       X11                         
##  language (EN)                        
##  collate  en_US.UTF-8                 
##  tz       UTC                         
##  date     2018-04-03</code></pre>
<pre><code>## Packages ------------------------------------------------------------------------------------------</code></pre>
<pre><code>##  package    * version   date       source        
##  ade4       * 1.7-10    2017-12-15 CRAN (R 3.4.3)
##  adegenet   * 2.1.1     2018-02-02 CRAN (R 3.4.3)
##  ape        * 5.0       2017-10-30 CRAN (R 3.4.3)
##  assertthat   0.2.0     2017-04-11 CRAN (R 3.4.3)
##  backports    1.1.2     2017-12-13 CRAN (R 3.4.3)
##  base       * 3.4.3     2018-02-21 local         
##  bindr        0.1       2016-11-13 CRAN (R 3.4.3)
##  bindrcpp     0.2       2017-06-17 CRAN (R 3.4.3)
##  boot         1.3-20    2017-07-30 CRAN (R 3.4.3)
##  cluster      2.0.6     2017-03-16 CRAN (R 3.4.3)
##  coda         0.19-1    2016-12-08 CRAN (R 3.4.3)
##  colorspace   1.3-2     2016-12-14 CRAN (R 3.4.3)
##  compiler     3.4.3     2018-02-21 local         
##  datasets   * 3.4.3     2018-02-21 local         
##  deldir       0.1-14    2017-04-22 CRAN (R 3.4.3)
##  devtools     1.13.5    2018-02-18 CRAN (R 3.4.3)
##  digest       0.6.15    2018-01-28 CRAN (R 3.4.3)
##  dplyr        0.7.4     2017-09-28 CRAN (R 3.4.3)
##  evaluate     0.10.1    2017-06-24 CRAN (R 3.4.3)
##  expm         0.999-2   2017-03-29 CRAN (R 3.4.3)
##  fastmatch    1.1-0     2017-01-28 CRAN (R 3.4.3)
##  gdata        2.18.0    2017-06-06 CRAN (R 3.4.3)
##  ggplot2      2.2.1     2016-12-30 CRAN (R 3.4.3)
##  glue         1.2.0     2017-10-29 CRAN (R 3.4.3)
##  gmodels      2.16.2    2015-07-22 CRAN (R 3.4.3)
##  graphics   * 3.4.3     2018-02-21 local         
##  grDevices  * 3.4.3     2018-02-21 local         
##  grid         3.4.3     2018-02-21 local         
##  gtable       0.2.0     2016-02-26 CRAN (R 3.4.3)
##  gtools       3.5.0     2015-05-29 CRAN (R 3.4.3)
##  htmltools    0.3.6     2017-04-28 CRAN (R 3.4.3)
##  httpuv       1.3.5     2017-07-04 CRAN (R 3.4.3)
##  igraph       1.1.2     2017-07-21 CRAN (R 3.4.3)
##  knitr        1.20      2018-02-20 CRAN (R 3.4.3)
##  lattice      0.20-35   2017-03-25 CRAN (R 3.4.3)
##  lazyeval     0.2.1     2017-10-29 CRAN (R 3.4.3)
##  LearnBayes   2.15      2014-05-29 CRAN (R 3.4.3)
##  magrittr     1.5       2014-11-22 CRAN (R 3.4.3)
##  MASS         7.3-48    2017-12-25 CRAN (R 3.4.3)
##  Matrix       1.2-12    2017-11-16 CRAN (R 3.4.3)
##  memoise      1.1.0     2017-04-21 CRAN (R 3.4.3)
##  methods    * 3.4.3     2018-02-21 local         
##  mgcv         1.8-23    2018-01-15 CRAN (R 3.4.3)
##  mime         0.5       2016-07-07 CRAN (R 3.4.3)
##  munsell      0.4.3     2016-02-13 CRAN (R 3.4.3)
##  nlme         3.1-131.1 2018-02-16 CRAN (R 3.4.3)
##  parallel     3.4.3     2018-02-21 local         
##  pegas      * 0.10      2017-05-03 CRAN (R 3.4.3)
##  permute      0.9-4     2016-09-09 CRAN (R 3.4.3)
##  phangorn     2.4.0     2018-02-15 CRAN (R 3.4.3)
##  pillar       1.1.0     2018-01-14 CRAN (R 3.4.3)
##  pkgconfig    2.0.1     2017-03-21 CRAN (R 3.4.3)
##  plyr         1.8.4     2016-06-08 CRAN (R 3.4.3)
##  poppr      * 2.6.1     2018-01-15 CRAN (R 3.4.3)
##  quadprog     1.5-5     2013-04-17 CRAN (R 3.4.3)
##  R6           2.2.2     2017-06-17 CRAN (R 3.4.3)
##  Rcpp         0.12.15   2018-01-20 CRAN (R 3.4.3)
##  reshape2     1.4.3     2017-12-11 CRAN (R 3.4.3)
##  rlang        0.2.0     2018-02-20 CRAN (R 3.4.3)
##  rmarkdown    1.8       2017-11-17 CRAN (R 3.4.3)
##  rprojroot    1.3-2     2018-01-03 CRAN (R 3.4.3)
##  scales       0.5.0     2017-08-24 CRAN (R 3.4.3)
##  seqinr       3.4-5     2017-08-01 CRAN (R 3.4.3)
##  shiny        1.0.5     2017-08-23 CRAN (R 3.4.3)
##  sp           1.2-7     2018-01-19 CRAN (R 3.4.3)
##  spData       0.2.7.4   2018-02-11 CRAN (R 3.4.3)
##  spdep        0.7-4     2017-11-22 CRAN (R 3.4.3)
##  splines      3.4.3     2018-02-21 local         
##  stats      * 3.4.3     2018-02-21 local         
##  stringi      1.1.6     2017-11-17 CRAN (R 3.4.3)
##  stringr      1.3.0     2018-02-19 CRAN (R 3.4.3)
##  tibble       1.4.2     2018-01-22 CRAN (R 3.4.3)
##  tools        3.4.3     2018-02-21 local         
##  utils      * 3.4.3     2018-02-21 local         
##  vegan        2.4-6     2018-01-24 CRAN (R 3.4.3)
##  withr        2.1.1     2017-12-19 CRAN (R 3.4.3)
##  xtable       1.8-2     2016-02-05 CRAN (R 3.4.3)
##  yaml         2.1.16    2017-12-12 CRAN (R 3.4.3)</code></pre>
</div>

<!-- </div>
</div>
</div>
 -->



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
