<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Detection of the signal of selection from genome scan. Population based analysis</title>

<script src="libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.5/css/spacelab.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="libs/navigation-1.1/tabsets.js"></script>
<!---

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
  padding-right: 50px;
  padding-left: -50px;
}
@media (max-width: 979px) {
  body {
    padding-top: 0;
  }
}

 offset scroll position for anchor links (for fixed navbar)  
@media (min-width: 980px) {
  .section h2 {
    padding-top: 52px;
    margin-top: -52px;
  }
  .section h3 {
    padding-top: 52px;
    margin-top: -52px;
  }
}


/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

</style>
-->



<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}
/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}
/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}
</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  { background-color: #f8f8f8; }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' || rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <!-- HOME -->
      <a class="navbar-brand" href="index.html"><b>Population Genetics in R</b></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <!-- Table of contents -->
        <li class="dropdown">
          <a href="chapters" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Users <span class="caret"></span></a>
          <ol class="dropdown-menu">
              <li><a href="GETSTARTED.html">Getting Started</a></li>
              <li class="divider"></li>
              <li class="dropdown-header">For microsatellite data:</li>
              <li><a href="startMicrosatellite.html">Basic statistics</a></li>
              <li><a href="2015-12-15-microsatellite-differentiation.html">Population differentiation</a></li>
              <li><a href="LMM-Genetic-Diversity.html">Comparing genetic diversity using mixed models</a></li>
              <li class="divider"></li>
              <li class="dropdown-header">For SNP data:</li>
              <!-- <li class="divider"></li> -->
              <li><a href="StartSNP.html">Basic statistics</a></li>
              <li><a href="DifferentiationSNP.html">Population differentiation</a></li>
              <li><a href="2015-05-18-Dist-SNP.html">Genetic distance (individual-based)</a></li>
              <li><a href="2016-01-26-SNP-selection.html">Signals of selection</a></li>
			  <li><a href="2016-12-13_MEM_outlier.html">Spatial outlier detection</a></li>
			  <li><a href="2018-03-27_RDA_GEA.html">Detecting multilocus adaptation using redundancy analysis</a></li>
              <!-- <li><a href="SNP.html">SNPs</a></li> -->
              <li class="divider"></li>
              <li class="dropdown-header">For sequence data:</li>
              <!-- <li class="divider"></li>   -->
              <li><a href="PopDiffSequenceData.html">Population differentiation</a></li>
          </ol>
        </li>
        <!-- Appendices dropdown -->
        <li class="dropdown">
          <a href="appendices" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Package Developers <span class="caret"></span></a>
          <ul class = "dropdown-menu" role = "menu">
           <li><a href = "PACKAGES.html">Population genetic packages in R</a></li>  
           <li><a href = "DATAFORMATS.html">Population genetic data types in R</a></li>
           <li><a href = "PACKAGE_TUTORIALS.html">Tutorials for developing R packages</a></li>
           <li class="divider"></li>  
           <li class="dropdown-header">Development Resources (external):</li>
           <li><a href = "http://r-pkgs.had.co.nz/">Hadley Wickham's guide for creating R packages</a></li>
           <li><a href = "https://github.com/jtleek/rpackages">Jeff Leek's R package development</a></li>
           <li><a href = "http://adv-r.had.co.nz/S4.html">S4 Classes in detail</a></li>
           <li><a href = "http://adv-r.had.co.nz/S3.html">S3 Classes in detail</a></li>
           <li><a href = "http://cran.r-project.org/doc/contrib/Genolini-S4tutorialV0-5en.pdf">A not so short introduction to S4</a></li>
          </ul>
        </li>
        <!-- Author button -->
        <li class="dropdown">
          <a href="about" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Contribute! <span class="caret"></span></a>
          <ul class = "dropdown-menu" role = "menu">
           <li><a href = "CONTRIBUTING.html">Contribution guidelines</a></li>  
           <li><a href = "CONTRIBUTING_WITH_GIT2R.html">Contributing with git2r</a></li> 
           <li class="divider"></li>  
           <li><a href="https://github.com/NESCent/popgenInfo">Source repository</a></li>  
		  </ul>
        </li>
        <li class = "button">
          <a href="LinksToResources.html">Useful Links</a>
        </li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<!-- <div class="container">
<div class="row">
<div class="span10 offset1"> -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Detection of the signal of selection from genome scan. Population based analysis</h1>

</div>

<div id="TOC">
<ul>
<li><a href="#introduction">Introduction</a></li>
<li><a href="#assumptions">Assumptions</a></li>
<li><a href="#data">Data</a></li>
<li><a href="#resourcespackages-required">Resources/Packages required</a><ul>
<li><a href="#section-1-load-the-data">Section 1: Load the data</a></li>
<li><a href="#section-2-pcadapt">Section 2: <em>PCAdapt</em></a></li>
<li><a href="#section-4-logistic-regression-linking-outliers-and-temperature">Section 4: Logistic regression : linking outliers and temperature</a></li>
</ul></li>
<li><a href="#conclusions">Conclusions</a><ul>
<li><a href="#whats-next">What’s next</a></li>
</ul></li>
<li><a href="#contributors">Contributors</a></li>
<li><a href="#session-information">Session Information</a></li>
</ul>
</div>

<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>The purpose of the vignette is to detect outlier SNP loci from a large number of SNPs genotyped for a large number of individuals. Individuals are organised in well defined populations. Then, we tested if allele counts in outlier loci are correlated to the temperature. We will first use a method based on principal component analysis (<em>PCAdapt</em>) (section 2) and a second method based on atypical values of Fst (<em>OutFLANK</em>; Whitlok and Lotterhos (2015); <a href="https://github.com/whitlock/OutFLANK" class="uri">https://github.com/whitlock/OutFLANK</a>) (section 3) to detect outliers loci. Then, we will test if the allele counts in outliers are correlated to the temperature using a Poisson regression (section 4).</p>
</div>
<div id="assumptions" class="section level1">
<h1>Assumptions</h1>
<ul>
<li>We assume that loci are neutral and we are looking for those are deviant from the neutral assumption.</li>
<li>We assume that loci correlated to the environment (temperature) are under selection.</li>
</ul>
</div>
<div id="data" class="section level1">
<h1>Data</h1>
<p>We will use simulated data provided by Katie Lotterhos (Lotterhos et al. in prep). The initial data set is constituted of 509 individuals, 19 populations and 3084 SNP. It has 2 extra-columns: Population and Temperature measures. SNP are in columns: each row contains one value at each SNP locus (separated by spaces or tabulations) corresponding to the number of reference alleles: 0 means zero copies of the reference allele, 1 means one copy of the reference allele (i.e. heterozygote), and 2 means two copies of the reference allele.</p>
</div>
<div id="resourcespackages-required" class="section level1">
<h1>Resources/Packages required</h1>
<ul>
<li><em>PCAdapt</em>: <a href="http://membres-timc.imag.fr/Michael.Blum/PCAdapt.html" class="uri">http://membres-timc.imag.fr/Michael.Blum/PCAdapt.html</a></li>
<li><em>OutFLANK</em>: <a href="https://github.com/whitlock/OutFLANK" class="uri">https://github.com/whitlock/OutFLANK</a></li>
</ul>
<p>Since <em>OutFLANK</em> is not yet on CRAN, you will need to install it from GitHub using the <em>devtools</em> R package. Additionally, <em>OutFLANK</em> depends on <em>qvalue</em> from <a href="http://www.bioconductor.org/packages/release/bioc/html/qvalue.html">Bioconductor</a>, which must be installed beforehand.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">install.packages</span>(<span class="st">&quot;devtools&quot;</span>)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw">library</span>(<span class="st">&quot;devtools&quot;</span>)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw">source</span>(<span class="st">&quot;https://bioconductor.org/biocLite.R&quot;</span>)</a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="kw">biocLite</span>(<span class="st">&quot;qvalue&quot;</span>)</a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="kw">install_github</span>(<span class="st">&quot;whitlock/OutFLANK&quot;</span>)</a></code></pre></div>
<p>Loading the required packages:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw">library</span>(<span class="st">&quot;pcadapt&quot;</span>)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw">library</span>(<span class="st">&quot;qvalue&quot;</span>)</a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="kw">library</span>(<span class="st">&quot;OutFLANK&quot;</span>)</a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="kw">library</span>(<span class="st">&quot;ggplot2&quot;</span>)</a></code></pre></div>
<div id="section-1-load-the-data" class="section level2">
<h2>Section 1: Load the data</h2>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">sel &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;data/SNPselection1.txt&quot;</span>, <span class="dt">head =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="kw">dim</span>(sel)</a></code></pre></div>
<pre><code>## [1]  509 3086</code></pre>
</div>
<div id="section-2-pcadapt" class="section level2">
<h2>Section 2: <em>PCAdapt</em></h2>
<p>To run the function <code>pcadapt()</code>, the user should specify the number K of principal components (PC) to work with: first perform with a large number of principal components (e.g. higher than the number of populations), then use the ‘scree plot’ to chose the value of K. It displays the percentage of variance that is explained by each PC. The recommended value of K corresponds to the largest value of K before the plateau of ‘scree plot’ is reached. Then for a given SNP, a statistical test to define the SNP as outlier or not is based on the “loadings” that are defined as the correlation between the SNP and the PCs. The statistic test for detecting outlier SNPs is the Mahalanobis distance (between the K correlations of the SNP and each axis and mean correlations) and, which scaled by a constant, should have a chi-square distribution with K degrees of freedom under the assumption that there are no outlier. By default P-values of SNPs with a minor allele frequency smaller than 0.05 are not computed. A Manhattan plot displays log10 of the p-values. It is also possible to check the distribution of the p-values using a Q-Q plot. The authors suggest to use false discovery rate (q-value) to provide a list of outliers.</p>
<blockquote>
<p>Note: <em>PCAdapt</em> expects the incoming matrix to have samples in columns and loci in rows. Because our data is the opposite, we need to transpose our data matrix with the <code>t()</code> function.</p>
</blockquote>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">genotype &lt;-<span class="st"> </span>sel[, <span class="dv">3</span><span class="op">:</span><span class="kw">ncol</span>(sel)]</a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="kw">dim</span>(genotype)</a></code></pre></div>
<pre><code>## [1]  509 3084</code></pre>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="co"># PCAdapt requires a pcadapt_class object. You can convert a matrix to </span></a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="co"># pcadapt_class with the read.pcadapt() function.</span></a>
<a class="sourceLine" id="cb7-3" data-line-number="3">pca_genotype &lt;-<span class="st"> </span><span class="kw">read.pcadapt</span>(<span class="kw">t</span>(genotype))</a></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">K &lt;-<span class="st"> </span><span class="dv">25</span></a>
<a class="sourceLine" id="cb8-2" data-line-number="2">x &lt;-<span class="st"> </span><span class="kw">pcadapt</span>(pca_genotype, <span class="dt">K =</span> K)</a>
<a class="sourceLine" id="cb8-3" data-line-number="3"><span class="kw">plot</span>(x, <span class="dt">option =</span> <span class="st">&quot;screeplot&quot;</span>) <span class="co"># 19 groups seems to be the correct value</span></a></code></pre></div>
<p><img src="2016-01-26-SNP-selection_files/figure-html/PCAdapt-1.png" width="960" /></p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw">plot</span>(x, <span class="dt">option =</span> <span class="st">&quot;scores&quot;</span>, <span class="dt">pop =</span> sel[, <span class="dv">1</span>]) <span class="co"># how populations are shared among the 19 groups</span></a></code></pre></div>
<p><img src="2016-01-26-SNP-selection_files/figure-html/PCAdapt-2.png" width="960" /></p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">K &lt;-<span class="st"> </span><span class="dv">19</span></a>
<a class="sourceLine" id="cb10-2" data-line-number="2">x &lt;-<span class="st"> </span><span class="kw">pcadapt</span>(pca_genotype, <span class="dt">K =</span> K, <span class="dt">min.maf =</span> <span class="dv">0</span>)</a>
<a class="sourceLine" id="cb10-3" data-line-number="3"></a>
<a class="sourceLine" id="cb10-4" data-line-number="4"><span class="kw">summary</span>(x) <span class="co"># numerical quantities obtained after performing a PCA</span></a></code></pre></div>
<pre><code>##                 Length Class  Mode   
## scores           9671  -none- numeric
## singular.values    19  -none- numeric
## loadings        58596  -none- numeric
## zscores         58596  -none- numeric
## af               3084  -none- numeric
## maf              3084  -none- numeric
## chi2.stat        3084  -none- numeric
## stat             3084  -none- numeric
## gif                 1  -none- numeric
## pvalues          3084  -none- numeric
## pass             3084  -none- numeric</code></pre>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="kw">plot</span>(x, <span class="dt">option =</span> <span class="st">&quot;manhattan&quot;</span>)</a></code></pre></div>
<p><img src="2016-01-26-SNP-selection_files/figure-html/PCAdapt-3.png" width="960" /></p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="kw">plot</span>(x, <span class="dt">option =</span> <span class="st">&quot;qqplot&quot;</span>, <span class="dt">threshold =</span> <span class="fl">0.1</span>)</a></code></pre></div>
<p><img src="2016-01-26-SNP-selection_files/figure-html/PCAdapt-4.png" width="960" /></p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="kw">plot</span>(x, <span class="dt">option =</span> <span class="st">&quot;stat.distribution&quot;</span>) <span class="co"># Distribution of Mahalanobis distances.</span></a></code></pre></div>
<p><img src="2016-01-26-SNP-selection_files/figure-html/PCAdapt-5.png" width="960" /></p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1">qval &lt;-<span class="st"> </span><span class="kw">qvalue</span>(x<span class="op">$</span>pvalues)<span class="op">$</span>qvalues</a>
<a class="sourceLine" id="cb15-2" data-line-number="2">alpha &lt;-<span class="st"> </span><span class="fl">0.1</span></a>
<a class="sourceLine" id="cb15-3" data-line-number="3">outliers_pcadapt &lt;-<span class="st"> </span><span class="kw">which</span>(qval <span class="op">&lt;</span><span class="st"> </span>alpha)</a>
<a class="sourceLine" id="cb15-4" data-line-number="4"><span class="kw">print</span>(outliers_pcadapt)</a></code></pre></div>
<pre><code>##  [1]  663  706  796  923 1155 1163 1347 1378 1639 1666 1825 2133 2556 2871</code></pre>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="kw">length</span>(outliers_pcadapt) <span class="co"># 14 outliers</span></a></code></pre></div>
<pre><code>## [1] 14</code></pre>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1">alpha &lt;-<span class="st"> </span><span class="fl">0.05</span> <span class="co"># use of a more stringent threshold to detect outliers</span></a>
<a class="sourceLine" id="cb19-2" data-line-number="2">outliers &lt;-<span class="st"> </span><span class="kw">which</span>(qval <span class="op">&lt;</span><span class="st"> </span>alpha)</a>
<a class="sourceLine" id="cb19-3" data-line-number="3"><span class="kw">print</span>(outliers)</a></code></pre></div>
<pre><code>##  [1]  663  706  796  923 1155 1163 1347 1378 1639 1666 1825 2133 2556 2871</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1"><span class="kw">length</span>(outliers) <span class="co"># 14 outliers</span></a></code></pre></div>
<pre><code>## [1] 14</code></pre>
<p>##section 3: <em>OutFLANK</em> (<a href="https://github.com/whitlock/OutFLANK" class="uri">https://github.com/whitlock/OutFLANK</a>)</p>
<p>A procedure to find Fst outliers based on an inferred distribution of neutral Fst: it uses likelihood on a trimmed distribution of Fst values to infer the distribution of Fst for neutral markers. This distribution is used to assign q-values to each locus to detect outliers loci potentially due to spatially heterogeneous selection. In practice, first the function <code>MakeDiploidFSTMat()</code> allows to calculate the appropriate input data frame among other parameters, Fst for each locus, and Fst without sampling size correction (<code>FSTNoCorr</code>). Then the function <code>OutFLANK()</code> estimate q-values and provides a list of outliers.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1">ind &lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="st">&quot;pop&quot;</span>, sel[, <span class="dv">1</span>]) <span class="co"># vector with the name of population</span></a>
<a class="sourceLine" id="cb23-2" data-line-number="2"></a>
<a class="sourceLine" id="cb23-3" data-line-number="3">locinames &lt;-<span class="st"> </span><span class="kw">as.character</span>(<span class="kw">seq</span>(<span class="kw">ncol</span>(genotype))) <span class="co"># vector with the name of loci</span></a>
<a class="sourceLine" id="cb23-4" data-line-number="4"></a>
<a class="sourceLine" id="cb23-5" data-line-number="5">FstDataFrame &lt;-<span class="st"> </span><span class="kw">MakeDiploidFSTMat</span>(genotype, locinames, ind)</a></code></pre></div>
<pre><code>## Calculating FSTs, may take a few minutes...</code></pre>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" data-line-number="1"><span class="kw">plot</span>(FstDataFrame<span class="op">$</span>FST, FstDataFrame<span class="op">$</span>FSTNoCorr, <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="op">-</span><span class="fl">0.01</span>,<span class="fl">0.3</span>), </a>
<a class="sourceLine" id="cb25-2" data-line-number="2">     <span class="dt">ylim =</span> <span class="kw">c</span>(<span class="op">-</span><span class="fl">0.01</span>, <span class="fl">0.3</span>), <span class="dt">pch =</span> <span class="dv">20</span>)</a>
<a class="sourceLine" id="cb25-3" data-line-number="3"><span class="kw">abline</span>(<span class="dv">0</span>, <span class="dv">1</span>) <span class="co"># Checking the effect of sample size on Fst since FSTCoCorr will be used in the follow</span></a></code></pre></div>
<p><img src="2016-01-26-SNP-selection_files/figure-html/outFLank-1.png" width="960" /></p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" data-line-number="1"><span class="kw">hist</span>(FstDataFrame<span class="op">$</span>FSTNoCorr) </a></code></pre></div>
<p><img src="2016-01-26-SNP-selection_files/figure-html/outFLank-2.png" width="960" /></p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" data-line-number="1">OF &lt;-<span class="st"> </span><span class="kw">OutFLANK</span>(FstDataFrame, <span class="dt">NumberOfSamples=</span><span class="dv">19</span>, <span class="dt">qthreshold =</span> <span class="fl">0.05</span>, </a>
<a class="sourceLine" id="cb27-2" data-line-number="2">               <span class="dt">RightTrimFraction =</span> <span class="fl">0.05</span>)</a>
<a class="sourceLine" id="cb27-3" data-line-number="3"></a>
<a class="sourceLine" id="cb27-4" data-line-number="4"><span class="co"># Plot the ditribution of Fst with the chi squared distribution</span></a>
<a class="sourceLine" id="cb27-5" data-line-number="5"><span class="kw">OutFLANKResultsPlotter</span>(OF, <span class="dt">withOutliers =</span> <span class="ot">TRUE</span>, <span class="dt">NoCorr =</span> <span class="ot">TRUE</span>, <span class="dt">Hmin =</span> <span class="fl">0.1</span>, </a>
<a class="sourceLine" id="cb27-6" data-line-number="6">                       <span class="dt">binwidth =</span> <span class="fl">0.005</span>, <span class="dt">Zoom =</span> <span class="ot">FALSE</span>, <span class="dt">RightZoomFraction =</span> <span class="fl">0.05</span>, </a>
<a class="sourceLine" id="cb27-7" data-line-number="7">                       <span class="dt">titletext =</span> <span class="ot">NULL</span>)</a></code></pre></div>
<p><img src="2016-01-26-SNP-selection_files/figure-html/outFLank-3.png" width="960" /></p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" data-line-number="1">outliers_OF &lt;-<span class="st"> </span>OF<span class="op">$</span>results<span class="op">$</span>LocusName[OF<span class="op">$</span>results<span class="op">$</span>OutlierFlag <span class="op">==</span><span class="st"> </span><span class="ot">TRUE</span>]</a>
<a class="sourceLine" id="cb28-2" data-line-number="2"><span class="kw">print</span>(outliers_OF)</a></code></pre></div>
<pre><code>##  [1] 663  706  923  1163 1378 1639 1666 1825 2133 2556 2871
## 3084 Levels: 1 10 100 1000 1001 1002 1003 1004 1005 1006 1007 1008 ... 999</code></pre>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb30-1" data-line-number="1"><span class="kw">length</span>(outliers_OF) <span class="co"># 11 outliers</span></a></code></pre></div>
<pre><code>## [1] 11</code></pre>
</div>
<div id="section-4-logistic-regression-linking-outliers-and-temperature" class="section level2">
<h2>Section 4: Logistic regression : linking outliers and temperature</h2>
<p>We aim to test if the outliers detected with <em>PCAdapt</em> and <em>OutFLANK</em> (663 706 923 1163 1378 1639 1666 1825 2133 2556 2871) are correlated to the temperature. We use a GLM with a binomial error to model the probability to have the allele “A” out of the two alleles. We derive the relation for some of the shared outliers. You can do the test for all.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb32-1" data-line-number="1"><span class="co"># We keep the outlier loci selected by both pcadapt and outflank</span></a>
<a class="sourceLine" id="cb32-2" data-line-number="2">outliers &lt;-<span class="st"> </span>outliers_pcadapt[outliers_pcadapt <span class="op">%in%</span><span class="st"> </span>outliers_OF]</a>
<a class="sourceLine" id="cb32-3" data-line-number="3"><span class="kw">length</span>(outliers) <span class="co"># 11 outliers</span></a></code></pre></div>
<pre><code>## [1] 11</code></pre>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb34-1" data-line-number="1">loc1 &lt;-<span class="st"> </span>genotype[, outliers[<span class="dv">1</span>]]</a>
<a class="sourceLine" id="cb34-2" data-line-number="2">temp &lt;-<span class="st"> </span>sel<span class="op">$</span>Temperature</a>
<a class="sourceLine" id="cb34-3" data-line-number="3">loc1temp &lt;-<span class="st"> </span><span class="kw">data.frame</span>(loc1, temp)</a>
<a class="sourceLine" id="cb34-4" data-line-number="4"> </a>
<a class="sourceLine" id="cb34-5" data-line-number="5">mod &lt;-<span class="st"> </span><span class="kw">glm</span>(<span class="kw">cbind</span>(loc1, <span class="dv">2</span> <span class="op">-</span><span class="st"> </span>loc1) <span class="op">~</span><span class="st"> </span>temp, <span class="dt">family =</span> binomial) </a>
<a class="sourceLine" id="cb34-6" data-line-number="6"><span class="kw">summary</span>(mod) <span class="co"># This locus is significantly correlated to temperature</span></a></code></pre></div>
<pre><code>## 
## Call:
## glm(formula = cbind(loc1, 2 - loc1) ~ temp, family = binomial)
## 
## Deviance Residuals: 
##      Min        1Q    Median        3Q       Max  
## -2.43957  -1.10354   0.09418   1.03271   2.62515  
## 
## Coefficients:
##             Estimate Std. Error z value Pr(&gt;|z|)    
## (Intercept) -0.02085    0.06770  -0.308    0.758    
## temp         0.27603    0.02603  10.606   &lt;2e-16 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for binomial family taken to be 1)
## 
##     Null deviance: 869.59  on 508  degrees of freedom
## Residual deviance: 738.61  on 507  degrees of freedom
## AIC: 1011.6
## 
## Number of Fisher Scoring iterations: 3</code></pre>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb36-1" data-line-number="1">loc2 &lt;-<span class="st"> </span>genotype[, outliers[<span class="dv">2</span>]]</a>
<a class="sourceLine" id="cb36-2" data-line-number="2">loc2temp &lt;-<span class="st"> </span><span class="kw">data.frame</span>(loc2, temp)</a>
<a class="sourceLine" id="cb36-3" data-line-number="3"><span class="kw">ggplot</span>(loc2temp, <span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">factor</span>(loc2), <span class="dt">y =</span> temp)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb36-4" data-line-number="4"><span class="st"> </span><span class="kw">geom_boxplot</span>() <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb36-5" data-line-number="5"><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;Major allele count&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb36-6" data-line-number="6"><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Temperature (Centigrade)&quot;</span>)</a></code></pre></div>
<p><img src="2016-01-26-SNP-selection_files/figure-html/Binomial%20regressions-1.png" width="960" /></p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb37-1" data-line-number="1">mod &lt;-<span class="st"> </span><span class="kw">glm</span>(<span class="kw">cbind</span>(loc2, <span class="dv">2</span> <span class="op">-</span><span class="st"> </span>loc2) <span class="op">~</span><span class="st"> </span>temp, <span class="dt">family =</span> binomial) </a>
<a class="sourceLine" id="cb37-2" data-line-number="2"><span class="kw">summary</span>(mod) <span class="co"># This locus is significantly correlated to temperature</span></a></code></pre></div>
<pre><code>## 
## Call:
## glm(formula = cbind(loc2, 2 - loc2) ~ temp, family = binomial)
## 
## Deviance Residuals: 
##      Min        1Q    Median        3Q       Max  
## -2.19071  -1.25625   0.06401   1.21800   2.20092  
## 
## Coefficients:
##             Estimate Std. Error z value Pr(&gt;|z|)    
## (Intercept) -0.02568    0.06552  -0.392    0.695    
## temp         0.19104    0.02411   7.923 2.32e-15 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for binomial family taken to be 1)
## 
##     Null deviance: 837.64  on 508  degrees of freedom
## Residual deviance: 769.88  on 507  degrees of freedom
## AIC: 1059.5
## 
## Number of Fisher Scoring iterations: 3</code></pre>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb39-1" data-line-number="1"><span class="co"># Test with a locus that is only selected by one method</span></a>
<a class="sourceLine" id="cb39-2" data-line-number="2">loc7 &lt;-<span class="st"> </span>genotype[, outliers_pcadapt[<span class="op">!</span>outliers_pcadapt <span class="op">%in%</span><span class="st"> </span>outliers_OF][<span class="dv">1</span>]]</a>
<a class="sourceLine" id="cb39-3" data-line-number="3">loc7temp &lt;-<span class="st"> </span><span class="kw">data.frame</span>(loc7, temp)</a>
<a class="sourceLine" id="cb39-4" data-line-number="4">  </a>
<a class="sourceLine" id="cb39-5" data-line-number="5"><span class="kw">ggplot</span>(loc2temp, <span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">factor</span>(loc7), <span class="dt">y =</span> temp)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb39-6" data-line-number="6"><span class="st"> </span><span class="kw">geom_boxplot</span>() <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb39-7" data-line-number="7"><span class="st"> </span><span class="kw">xlab</span>(<span class="st">&quot;Major allele count&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb39-8" data-line-number="8"><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;Temperature (Centigrade)&quot;</span>)</a></code></pre></div>
<p><img src="2016-01-26-SNP-selection_files/figure-html/Binomial%20regressions-2.png" width="960" /></p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb40-1" data-line-number="1">mod &lt;-<span class="st"> </span><span class="kw">glm</span>(<span class="kw">cbind</span>(loc7, <span class="dv">2</span> <span class="op">-</span><span class="st"> </span>loc7) <span class="op">~</span><span class="st"> </span>temp, <span class="dt">family =</span> binomial)</a>
<a class="sourceLine" id="cb40-2" data-line-number="2"><span class="kw">summary</span>(mod) <span class="co">## This locus is not significantly correlated to temperature</span></a></code></pre></div>
<pre><code>## 
## Call:
## glm(formula = cbind(loc7, 2 - loc7) ~ temp, family = binomial)
## 
## Deviance Residuals: 
##      Min        1Q    Median        3Q       Max  
## -1.84798  -1.59313  -0.06144   1.53467   1.73831  
## 
## Coefficients:
##             Estimate Std. Error z value Pr(&gt;|z|)  
## (Intercept)  0.06980    0.06346   1.100    0.271  
## temp        -0.04202    0.02243  -1.874    0.061 .
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for binomial family taken to be 1)
## 
##     Null deviance: 802.15  on 508  degrees of freedom
## Residual deviance: 798.62  on 507  degrees of freedom
## AIC: 1106.2
## 
## Number of Fisher Scoring iterations: 3</code></pre>
</div>
</div>
<div id="conclusions" class="section level1">
<h1>Conclusions</h1>
<ul>
<li><em>PCAdapt</em> detected 14 outliers</li>
<li><em>OutFLANK</em> detected 11 outliers</li>
<li>11 outliers were detected in common</li>
<li>Those common outliers are all significantly correlated with the temperatures as shown by Poisson regression. Such outliers can be potentially involved in local adaptation due to temperatures. However more analysis are necessary to confirm that they are outliers, and which allele is potentially under selection.</li>
</ul>
<div id="whats-next" class="section level2">
<h2>What’s next</h2>
<p>Information on further analysis that could be done as LFFM (package <em>LEA</em>..).</p>
</div>
</div>
<div id="contributors" class="section level1">
<h1>Contributors</h1>
<ul>
<li>Stephanie Manel (EPHE) (Author)</li>
<li>Alicia Dalongeville (CEFE) (Author)</li>
<li>Zhian Kamvar (reviewer)</li>
</ul>
<p>#References</p>
<p>Duforet-Frebourg N, K Luu, G Laval, E Bazin, MGB Blum. (2016) Detecting genomic signatures of natural selection with principal component analysis: application to the 1000 Genomes data. Molecular Biology and Evolution (<a href="http://arxiv.org/abs/1504.04543" class="uri">http://arxiv.org/abs/1504.04543</a>)</p>
<p>Whitlock MC, Lotterhos KE (2015) Reliable Detection of Loci Responsible for Local Adaptation: Inference of a Null Model through Trimming the Distribution of Fst. The American Naturalist 186, S24-S36. <a href="http://www.jstor.org/stable/10.1086/682949" class="uri">http://www.jstor.org/stable/10.1086/682949</a></p>
</div>
<div id="session-information" class="section level1">
<h1>Session Information</h1>
<p>This shows us useful information for reproducibility. Of particular importance are the versions of R and the packages used to create this workflow. It is considered good practice to record this information with every analysis.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb42-1" data-line-number="1"><span class="kw">options</span>(<span class="dt">width =</span> <span class="dv">100</span>)</a>
<a class="sourceLine" id="cb42-2" data-line-number="2">devtools<span class="op">::</span><span class="kw">session_info</span>()</a></code></pre></div>
<pre><code>## ─ Session info ───────────────────────────────────────────────────────────────────────────────────
##  setting  value                       
##  version  R version 3.6.0 (2019-04-26)
##  os       Debian GNU/Linux 9 (stretch)
##  system   x86_64, linux-gnu           
##  ui       X11                         
##  language (EN)                        
##  collate  en_US.UTF-8                 
##  ctype    en_US.UTF-8                 
##  tz       Etc/UTC                     
##  date     2019-05-11                  
## 
## ─ Packages ───────────────────────────────────────────────────────────────────────────────────────
##  package     * version  date       lib source                            
##  ape           5.3      2019-03-17 [1] CRAN (R 3.6.0)                    
##  assertthat    0.2.1    2019-03-21 [1] CRAN (R 3.6.0)                    
##  backports     1.1.4    2019-04-10 [1] CRAN (R 3.6.0)                    
##  callr         3.2.0    2019-03-15 [1] CRAN (R 3.6.0)                    
##  cli           1.1.0    2019-03-19 [1] CRAN (R 3.6.0)                    
##  cluster       2.0.8    2019-04-05 [2] CRAN (R 3.6.0)                    
##  colorspace    1.4-1    2019-03-18 [1] CRAN (R 3.6.0)                    
##  crayon        1.3.4    2017-09-16 [1] CRAN (R 3.6.0)                    
##  data.table    1.12.2   2019-04-07 [1] CRAN (R 3.6.0)                    
##  DEoptimR      1.0-8    2016-11-19 [1] CRAN (R 3.6.0)                    
##  desc          1.2.0    2018-05-01 [1] CRAN (R 3.6.0)                    
##  devtools      2.0.2    2019-04-08 [1] CRAN (R 3.6.0)                    
##  digest        0.6.18   2018-10-10 [1] CRAN (R 3.6.0)                    
##  dplyr         0.8.0.1  2019-02-15 [1] CRAN (R 3.6.0)                    
##  evaluate      0.13     2019-02-12 [1] CRAN (R 3.6.0)                    
##  fit.models    0.5-14   2017-04-06 [1] CRAN (R 3.6.0)                    
##  fs            1.3.1    2019-05-06 [1] CRAN (R 3.6.0)                    
##  ggplot2     * 3.1.1    2019-04-07 [1] CRAN (R 3.6.0)                    
##  glue          1.3.1    2019-03-12 [1] CRAN (R 3.6.0)                    
##  gtable        0.3.0    2019-03-25 [1] CRAN (R 3.6.0)                    
##  htmltools     0.3.6    2017-04-28 [1] CRAN (R 3.6.0)                    
##  htmlwidgets   1.3      2018-09-30 [1] CRAN (R 3.6.0)                    
##  httr          1.4.0    2018-12-11 [1] CRAN (R 3.6.0)                    
##  jsonlite      1.6      2018-12-07 [1] CRAN (R 3.6.0)                    
##  knitr         1.22     2019-03-08 [1] CRAN (R 3.6.0)                    
##  labeling      0.3      2014-08-23 [1] CRAN (R 3.6.0)                    
##  lattice       0.20-38  2018-11-04 [2] CRAN (R 3.6.0)                    
##  lazyeval      0.2.2    2019-03-15 [1] CRAN (R 3.6.0)                    
##  magrittr      1.5      2014-11-22 [1] CRAN (R 3.6.0)                    
##  MASS          7.3-51.4 2019-03-31 [2] CRAN (R 3.6.0)                    
##  Matrix        1.2-17   2019-03-22 [2] CRAN (R 3.6.0)                    
##  memoise       1.1.0    2017-04-21 [1] CRAN (R 3.6.0)                    
##  mgcv          1.8-28   2019-03-21 [2] CRAN (R 3.6.0)                    
##  munsell       0.5.0    2018-06-12 [1] CRAN (R 3.6.0)                    
##  mvtnorm       1.0-10   2019-03-05 [1] CRAN (R 3.6.0)                    
##  nlme          3.1-139  2019-04-09 [2] CRAN (R 3.6.0)                    
##  OutFLANK    * 0.2      2019-05-10 [1] Github (whitlock/OutFLANK@8ec7d50)
##  pcadapt     * 4.1.0    2019-02-27 [1] CRAN (R 3.6.0)                    
##  pcaPP         1.9-73   2018-01-14 [1] CRAN (R 3.6.0)                    
##  permute       0.9-5    2019-03-12 [1] CRAN (R 3.6.0)                    
##  pillar        1.3.1    2018-12-15 [1] CRAN (R 3.6.0)                    
##  pinfsc50      1.1.0    2016-12-02 [1] CRAN (R 3.6.0)                    
##  pkgbuild      1.0.3    2019-03-20 [1] CRAN (R 3.6.0)                    
##  pkgconfig     2.0.2    2018-08-16 [1] CRAN (R 3.6.0)                    
##  pkgload       1.0.2    2018-10-29 [1] CRAN (R 3.6.0)                    
##  plotly        4.9.0    2019-04-10 [1] CRAN (R 3.6.0)                    
##  plyr          1.8.4    2016-06-08 [1] CRAN (R 3.6.0)                    
##  prettyunits   1.0.2    2015-07-13 [1] CRAN (R 3.6.0)                    
##  processx      3.3.1    2019-05-08 [1] CRAN (R 3.6.0)                    
##  ps            1.3.0    2018-12-21 [1] CRAN (R 3.6.0)                    
##  purrr         0.3.2    2019-03-15 [1] CRAN (R 3.6.0)                    
##  qvalue      * 2.16.0   2019-05-02 [1] Bioconductor                      
##  R6            2.4.0    2019-02-14 [1] CRAN (R 3.6.0)                    
##  Rcpp          1.0.1    2019-03-17 [1] CRAN (R 3.6.0)                    
##  remotes       2.0.4    2019-04-10 [1] CRAN (R 3.6.0)                    
##  reshape2      1.4.3    2017-12-11 [1] CRAN (R 3.6.0)                    
##  rlang         0.3.4    2019-04-07 [1] CRAN (R 3.6.0)                    
##  rmarkdown     1.12     2019-03-14 [1] CRAN (R 3.6.0)                    
##  robust        0.4-18   2017-04-27 [1] CRAN (R 3.6.0)                    
##  robustbase    0.93-4   2019-03-19 [1] CRAN (R 3.6.0)                    
##  rprojroot     1.3-2    2018-01-03 [1] CRAN (R 3.6.0)                    
##  rrcov         1.4-7    2018-11-15 [1] CRAN (R 3.6.0)                    
##  RSpectra      0.14-0   2019-04-04 [1] CRAN (R 3.6.0)                    
##  scales        1.0.0    2018-08-09 [1] CRAN (R 3.6.0)                    
##  sessioninfo   1.1.1    2018-11-05 [1] CRAN (R 3.6.0)                    
##  stringi       1.4.3    2019-03-12 [1] CRAN (R 3.6.0)                    
##  stringr       1.4.0    2019-02-10 [1] CRAN (R 3.6.0)                    
##  testthat      2.1.1    2019-04-23 [1] CRAN (R 3.6.0)                    
##  tibble        2.1.1    2019-03-16 [1] CRAN (R 3.6.0)                    
##  tidyr         0.8.3    2019-03-01 [1] CRAN (R 3.6.0)                    
##  tidyselect    0.2.5    2018-10-11 [1] CRAN (R 3.6.0)                    
##  usethis       1.5.0    2019-04-07 [1] CRAN (R 3.6.0)                    
##  vcfR          1.8.0    2018-04-17 [1] CRAN (R 3.6.0)                    
##  vegan         2.5-4    2019-02-04 [1] CRAN (R 3.6.0)                    
##  viridisLite   0.3.0    2018-02-01 [1] CRAN (R 3.6.0)                    
##  withr         2.1.2    2018-03-15 [1] CRAN (R 3.6.0)                    
##  xfun          0.6      2019-04-02 [1] CRAN (R 3.6.0)                    
##  yaml          2.2.0    2018-07-25 [1] CRAN (R 3.6.0)                    
## 
## [1] /usr/local/lib/R/site-library
## [2] /usr/local/lib/R/library</code></pre>
</div>

<!-- </div>
</div>
</div>
 -->



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
