<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Data classes</title>

<script src="libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.5/css/spacelab.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="libs/navigation-1.1/tabsets.js"></script>
<!---

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
  padding-right: 50px;
  padding-left: -50px;
}
@media (max-width: 979px) {
  body {
    padding-top: 0;
  }
}

 offset scroll position for anchor links (for fixed navbar)  
@media (min-width: 980px) {
  .section h2 {
    padding-top: 52px;
    margin-top: -52px;
  }
  .section h3 {
    padding-top: 52px;
    margin-top: -52px;
  }
}


/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

</style>
-->



<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}
/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}
/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}
</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>




<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <!-- HOME -->
      <a class="navbar-brand" href="index.html"><b>Population Genetics in R</b></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <!-- Table of contents -->
        <li class="dropdown">
          <a href="chapters" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Users <span class="caret"></span></a>
          <ol class="dropdown-menu">
              <li><a href="GETSTARTED.html">Getting Started</a></li>
              <li class="divider"></li>
              <li class="dropdown-header">For microsatellite data:</li>
              <li><a href="startMicrosatellite.html">Basic statistics</a></li>
              <li><a href="2015-12-15-microsatellite-differentiation.html">Population differentiation</a></li>
              <li><a href="LMM-Genetic-Diversity.html">Comparing genetic diversity using mixed models</a></li>
              <li class="divider"></li>
              <li class="dropdown-header">For SNP data:</li>
              <!-- <li class="divider"></li> -->
              <li><a href="StartSNP.html">Basic statistics</a></li>
              <li><a href="DifferentiationSNP.html">Population differentiation</a></li>
              <li><a href="2015-05-18-Dist-SNP.html">Genetic distance (individual-based)</a></li>
              <li><a href="2016-01-26-SNP-selection.html">Signals of selection</a></li>
			  <li><a href="2016-12-13_MEM_outlier.html">Spatial outlier detection</a></li>
			  <li><a href="2018-03-27_RDA_GEA.html">Detecting multilocus adaptation using redundancy analysis</a></li>
              <!-- <li><a href="SNP.html">SNPs</a></li> -->
              <li class="divider"></li>
              <li class="dropdown-header">For sequence data:</li>
              <!-- <li class="divider"></li>   -->
              <li><a href="PopDiffSequenceData.html">Population differentiation</a></li>
          </ol>
        </li>
        <!-- Appendices dropdown -->
        <li class="dropdown">
          <a href="appendices" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Package Developers <span class="caret"></span></a>
          <ul class = "dropdown-menu" role = "menu">
           <li><a href = "PACKAGES.html">Population genetic packages in R</a></li>  
           <li><a href = "DATAFORMATS.html">Population genetic data types in R</a></li>
           <li><a href = "PACKAGE_TUTORIALS.html">Tutorials for developing R packages</a></li>
           <li class="divider"></li>  
           <li class="dropdown-header">Development Resources (external):</li>
           <li><a href = "http://r-pkgs.had.co.nz/">Hadley Wickham's guide for creating R packages</a></li>
           <li><a href = "https://github.com/jtleek/rpackages">Jeff Leek's R package development</a></li>
           <li><a href = "http://adv-r.had.co.nz/S4.html">S4 Classes in detail</a></li>
           <li><a href = "http://adv-r.had.co.nz/S3.html">S3 Classes in detail</a></li>
           <li><a href = "http://cran.r-project.org/doc/contrib/Genolini-S4tutorialV0-5en.pdf">A not so short introduction to S4</a></li>
          </ul>
        </li>
        <!-- Author button -->
        <li class="dropdown">
          <a href="about" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Contribute! <span class="caret"></span></a>
          <ul class = "dropdown-menu" role = "menu">
           <li><a href = "CONTRIBUTING.html">Contribution guidelines</a></li>  
           <li><a href = "CONTRIBUTING_WITH_GIT2R.html">Contributing with git2r</a></li> 
           <li class="divider"></li>  
           <li><a href="https://github.com/NESCent/popgenInfo">Source repository</a></li>  
		  </ul>
        </li>
        <li class = "button">
          <a href="LinksToResources.html">Useful Links</a>
        </li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<!-- <div class="container">
<div class="row">
<div class="span10 offset1"> -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Data classes</h1>

</div>


<p>The following table describes specialized objects to store data represented in population genetics packages. Conversion between all types is possible.</p>
<p>Anyone developing a package for population genetic analysis is encouraged to use or build upon these data structures. If a new data structure is needed, please provide a conversion method to one or more of the classes listed below.</p>
<table>
<colgroup>
<col width="46%" />
<col width="26%" />
<col width="27%" />
</colgroup>
<thead>
<tr class="header">
<th>Class {type} (<em>package</em>)</th>
<th>Strengths</th>
<th>Weaknesses</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>DNAbin</strong> {S3} (<em>ape</em>)</td>
<td>stores all sets of sequences (aligned or not)</td>
<td>less compact than 2-bit coding (but by a factor 4 at most)</td>
</tr>
<tr class="even">
<td></td>
<td>uses matrices (aligned) or lists so usual R’s commands (<code>names</code>, <code>rownames</code>, <code>[</code>, <code>[[</code>, <code>$</code>) can be used</td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td>many <code>as.DNAbin</code> methods in ape (inc. from BioConductor)</td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td>efficient functions in ape (<code>dist.dna</code>, <code>seg.sites</code>, <code>base.freq</code>, <code>read.FASTA</code>) and in pegas (<code>haplotype</code>)</td>
<td></td>
</tr>
<tr class="odd">
<td><strong>loci</strong> {S3} (<em>pegas</em>)</td>
<td>low memory usage</td>
<td>not really appropriate for some analyses (e.g., multivariate analyses)</td>
</tr>
<tr class="even">
<td></td>
<td>all levels of ploidy and any number of alleles</td>
<td>needs to improve the treatment of NA’s (especially when data are read with read.vcf()</td>
</tr>
<tr class="odd">
<td></td>
<td>genotypes can be phased</td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td>any kind of individual data can be associated in the data frame</td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td>efficient to compute genotype and allele frequencies</td>
<td></td>
</tr>
<tr class="even">
<td><strong>genind</strong> {S4} (<em>adegenet</em>)</td>
<td>stores allelic counts; ideal for multivariate analyses</td>
<td>requires more memory</td>
</tr>
<tr class="odd">
<td> </td>
<td>additional slots for individual data</td>
<td>less efficient to compute frequencies</td>
</tr>
<tr class="even">
<td></td>
<td>additional slot for population strata</td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td>all levels of ploidy</td>
<td></td>
</tr>
<tr class="even">
<td><strong>genpop</strong> {S4} (<em>adegenet</em>)</td>
<td>equivalent to genind at group level; ideal for multivariate analysis</td>
<td>requires more memory</td>
</tr>
<tr class="odd">
<td><strong>genlight</strong> {S4} (<em>adegenet</em>)</td>
<td>stores binary SNPs using bit-level coding; very memory efficient</td>
<td>more computationally intensive to handle; less functionalities</td>
</tr>
<tr class="even">
<td></td>
<td>additional slots for individual data and population strata</td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td>all levels of ploidy</td>
<td>assumes bi-allelic loci</td>
</tr>
<tr class="even">
<td><strong>genclone</strong> {S4} (<em>poppr</em>)</td>
<td>inherits genind object; gains all advantages</td>
<td>all the same weaknesses plus slightly more memory</td>
</tr>
<tr class="odd">
<td></td>
<td>stores multilocus genotype/lineage definitions (<code>@mlg</code> slot) for clonal populations</td>
<td></td>
</tr>
<tr class="even">
<td><strong>snpclone</strong> {S4} (<em>poppr</em>)</td>
<td>inherits genlight object; gains all advantages</td>
<td>all the same weaknesses plus slightly more memory</td>
</tr>
<tr class="odd">
<td></td>
<td>stores multilocus genotype/lineage definitions (<code>@mlg</code> slot) for clonal populations</td>
<td></td>
</tr>
<tr class="even">
<td><strong>genambig</strong> {S4} (<em>polysat</em>)</td>
<td>stores microsatellite data with ambiguous ploidy</td>
<td>does not handle any other data type</td>
</tr>
<tr class="odd">
<td></td>
<td>exports to genpop objects</td>
<td>cannot easily be transferred to any other object</td>
</tr>
<tr class="even">
<td><strong>phyDat</strong> {S3} (<em>phangorn</em>)</td>
<td>very general inspired by R <code>data.frame</code>, <code>factor</code> and <code>contrasts</code>, can contain any discrete data type; nucleotides, amino acids and codons have some more support</td>
<td>designed having phylogenetic analysis in mind; requires alignments, where all sequences have same length</td>
</tr>
<tr class="odd">
<td></td>
<td>can be converted to and from <code>DNAbin</code> objects (<code>as.DNAbin</code> / <code>as.phyDat</code>)</td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td>a few generic functions work on it: <code>c</code>, <code>unique</code>, <code>subset</code> and utility functions <code>baseFreq</code>, <code>allSitePattern</code>, etc.</td>
<td>data are not necessarily very memory efficient (as integer + contrast matrix), but stores only unique site patterns and their weights (as double)</td>
</tr>
<tr class="odd">
<td></td>
<td>“efficient” maximum likelihood, maximum parsimony and distance functions in phangorn</td>
<td></td>
</tr>
<tr class="even">
<td><strong>gtype</strong> {S3} (<em>strataG</em>)</td>
<td>a simple R <code>list</code> containing a <code>matrix</code> where the first column is a stratification scheme and columns afterward are either haplotypes or diploid loci. If haploid data, the <code>gtype</code> object can also contain a list of DNA sequences.</td>
<td>Can likely be made more efficient in terms of storage and preprocessing for other analytical routines in package</td>
</tr>
<tr class="odd">
<td></td>
<td>can be converted to <code>data.frame</code> or <code>matrix</code> with appropriate <code>as.</code> functions.</td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td>has manipulation functions like <code>subset</code> which will select certain strata and/or loci, <code>merge</code> to combine mulitple <code>gtypes</code>, and <code>summary</code>.</td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td>can create input files for Genepop, STRUCTURE, fastsimcoal, Arlequin, MEGA, and PHASE</td>
<td></td>
</tr>
<tr class="even">
<td><strong>multiDNA</strong> {S4} (<em>apex</em>)</td>
<td>stores multiple <code>DNAbin</code> objects from <em>ape</em></td>
<td></td>
</tr>
<tr class="odd">
<td><strong>multiPhyDat</strong> {S4} (<em>apex</em>)</td>
<td>stores multiple <code>phyDat</code> objects from <em>phangorn</em></td>
<td></td>
</tr>
</tbody>
</table>

<!-- </div>
</div>
</div>
 -->



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
