<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />



<title>Data classes</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/spacelab.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>
<!---

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
  padding-right: 50px;
  padding-left: -50px;
}
@media (max-width: 979px) {
  body {
    padding-top: 0;
  }
}

 offset scroll position for anchor links (for fixed navbar)  
@media (min-width: 980px) {
  .section h2 {
    padding-top: 52px;
    margin-top: -52px;
  }
  .section h3 {
    padding-top: 52px;
    margin-top: -52px;
  }
}


/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

</style>
-->



<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}
/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}
/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}
</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>




</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <!-- HOME -->
      <a class="navbar-brand" href="index.html"><b>Population Genetics in R</b></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <!-- Table of contents -->
        <li class="dropdown">
          <a href="chapters" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Users <span class="caret"></span></a>
          <ol class="dropdown-menu">
              <li><a href="GETSTARTED.html">Getting Started</a></li>
              <li class="divider"></li>
              <li class="dropdown-header">For SNP data:</li>
              <!-- <li class="divider"></li> -->
              <li><a href="StartSNP.html">Basic statistics</a></li>
              <li><a href="DifferentiationSNP.html">Population differentiation</a></li>
              <!-- <li><a href="SNP.html">SNPs</a></li> -->
              <li class="divider"></li>
              <li class="dropdown-header">For sequence data:</li>
              <!-- <li class="divider"></li>   -->
              <li><a href="SummarySequenceData.html">Summary statistics</a></li>
              <li><a href="PopDiffSequenceData.html">Population differentiation</a></li>
          </ol>
        </li>
        <!-- Appendices dropdown -->
        <li class="dropdown">
          <a href="appendices" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Package Developers <span class="caret"></span></a>
          <ul class = "dropdown-menu", role = "menu">
           <li><a href = "DEVELOPER.html">Introduction</a></li>  
          </ul>
        </li>
        <!-- Author button -->
        <li class="dropdown">
          <a href="about" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Contribute! <span class="caret"></span></a>
          <ul class = "dropdown-menu", role = "menu">
           <li><a href = "CONTRIBUTING.html">How to contribute</a></li>  
           <li><a href="https://github.com/NESCent/popgenInfo">Source repository</a></li>  
		  </ul>
        </li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<!-- <div class="container">
<div class="row">
<div class="span10 offset1"> -->

<div id="header">
<h1 class="title">Data classes</h1>
</div>


<table>
<thead>
<tr class="header">
<th align="left">Class (package)</th>
<th align="left">Strengths</th>
<th align="left">Weaknesses</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><strong>DNAbin</strong> (ape)</td>
<td align="left">stores all sets of sequences (aligned or not)</td>
<td align="left">less compact than 2-bit coding (but by a factor 4 at most)</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">uses matrices (aligned) or lists so usual R’s commands (<code>names</code>, <code>rownames</code>, <code>[</code>, <code>[[</code>, <code>$</code>) can be used</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">many <code>as.DNAbin</code> methods in ape (inc. from BioConductor)</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">efficient functions in ape (<code>dist.dna</code>, <code>seg.sites</code>, <code>base.freq</code>, <code>read.FASTA</code>) and in pegas (<code>haplotype</code>)</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left"><strong>loci</strong> (pegas)</td>
<td align="left">low memory usage</td>
<td align="left">not really appropriate for some analyses (e.g., multivariate analyses)</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">all levels of ploidy and any number of alleles</td>
<td align="left">needs to improve the treatment of NA’s (especially when data are read with read.vcf()</td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">genotypes can be phased</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">any kind of individual data can be associated in the data frame</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">efficient to compute genotype and allele frequencies</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left"><strong>genind</strong> (adegenet)</td>
<td align="left">stores allelic frequencies; ideal for multivariate analyses</td>
<td align="left">requires more memory</td>
</tr>
<tr class="odd">
<td align="left"> </td>
<td align="left">additional slots for individual data</td>
<td align="left">less efficient to compute frequencies</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">all levels of ploidy</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left"><strong>genpop</strong> (adegenet)</td>
<td align="left">equivalent to genind at group level; ideal for multivariate analysis</td>
<td align="left">requires more memory</td>
</tr>
<tr class="even">
<td align="left"><strong>genlight</strong> (adegenet)</td>
<td align="left">stores binary SNPs using bit-level coding; very memory efficient</td>
<td align="left">more computationally intensive to handle; less functionalities</td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">all levels of ploidy</td>
<td align="left">assumes bi-allelic loci</td>
</tr>
<tr class="even">
<td align="left"><strong>genclone</strong> (poppr)</td>
<td align="left">stores population metadata (<code>@hierarchy</code> slot); users don’t have keep track of multiple objects for the same data</td>
<td align="left">inherits genind object; all the same weaknesses plus slightly more memory</td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">stores multilocus genotype definitions (<code>@mlg</code> slot) for clonal populations</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left"><strong>genambig</strong> (polysat)</td>
<td align="left">stores microsatellite data with ambiguous ploidy</td>
<td align="left">does not handle any other data type</td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">exports to genpop objects</td>
<td align="left">cannot easily be transferred to any other object</td>
</tr>
<tr class="even">
<td align="left"><strong>phyDat</strong> (phangorn)</td>
<td align="left">very general inspired by R <code>data.frame</code>, <code>factor</code> and <code>contrasts</code>, can contain any discrete data type; nucleotides, amino acids and codons have some more support</td>
<td align="left">designed having phylogenetic analysis in mind; requires alignments, where all sequences have same length</td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">can be converted to and from <code>DNAbin</code> objects (<code>as.DNAbin</code> / <code>as.phyDat</code>)</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">a few generic functions work on it: <code>c</code>, <code>unique</code>, <code>subset</code> and utility functions <code>baseFreq</code>, <code>allSitePattern</code>, etc.</td>
<td align="left">data are not necessarily very memory efficient (as integer + contrast matrix), but stores only unique site patterns and their weights (as double)</td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">“efficient” maximum likelihood, maximum parsimony and distance functions in phangorn</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left"><strong>gtype</strong> (strataG)</td>
<td align="left">a simple R <code>list</code> containing a <code>matrix</code> where the first column is a stratification scheme and columns afterward are either haplotypes or diploid loci. If haploid data, the <code>gtype</code> object can also contain a list of DNA sequences.</td>
<td align="left">Can likely be made more efficient in terms of storage and preprocessing for other analytical routines in package</td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">can be converted to <code>data.frame</code> or <code>matrix</code> with appropriate <code>as.</code> functions.</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">has manipulation functions like <code>subset</code> which will select certain strata and/or loci, <code>merge</code> to combine mulitple <code>gtypes</code>, and <code>summary</code>.</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">can create input files for Genepop, STRUCTURE, fastsimcoal, Arlequin, MEGA, and PHASE</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left"><strong>vareff</strong> (VarEff)</td>
<td align="left">this format is the only input format for the package VarEff, which is the only R package I could find to estimate the effective population sizes from present to ancestral time using microsatellite markers under the coalescence - AKA skyline plots</td>
<td align="left">it is very difficult to convert any genotype data format into the vareff format</td>
</tr>
<tr class="odd">
<td align="left"><strong>other</strong></td>
<td align="left"></td>
<td align="left"></td>
</tr>
</tbody>
</table>

<!-- </div>
</div>
</div>
 -->

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
