<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="author" content="Margarita M. López-Uribe" />

<meta name="date" content="2015-03-19" />

<title>Population Differentiation for Sequence Data</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/spacelab.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>
<!---

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
  padding-right: 50px;
  padding-left: -50px;
}
@media (max-width: 979px) {
  body {
    padding-top: 0;
  }
}

 offset scroll position for anchor links (for fixed navbar)  
@media (min-width: 980px) {
  .section h2 {
    padding-top: 52px;
    margin-top: -52px;
  }
  .section h3 {
    padding-top: 52px;
    margin-top: -52px;
  }
}


/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

</style>
-->



<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}
/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}
/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}
</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #f8f8f8; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
pre, code { background-color: #f8f8f8; }
code > span.kw { color: #204a87; font-weight: bold; }
code > span.dt { color: #204a87; }
code > span.dv { color: #0000cf; }
code > span.bn { color: #0000cf; }
code > span.fl { color: #0000cf; }
code > span.ch { color: #4e9a06; }
code > span.st { color: #4e9a06; }
code > span.co { color: #8f5902; font-style: italic; }
code > span.ot { color: #8f5902; }
code > span.al { color: #ef2929; }
code > span.fu { color: #000000; }
code > span.er { font-weight: bold; }
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <!-- HOME -->
      <a class="navbar-brand" href="index.html"><b>Population Genetics in R</b></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <!-- Table of contents -->
        <li class="dropdown">
          <a href="chapters" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Users <span class="caret"></span></a>
          <ol class="dropdown-menu">
              <li><a href="GETSTARTED.html">Getting Started</a></li>
              <li class="divider"></li>
              <li class="dropdown-header">For microsatellite data:</li>
              <li><a href="startMicrosatellite.html">Basic Statistics</a></li>
              <li class="divider"></li>
              <li class="dropdown-header">For SNP data:</li>
              <!-- <li class="divider"></li> -->
              <li><a href="StartSNP.html">Basic statistics</a></li>
              <li><a href="DifferentiationSNP.html">Population differentiation</a></li>
              <li><a href="2015-05-18-Dist-SNP.html">Genetic Distance (individual-based)</a></li>
              <!-- <li><a href="SNP.html">SNPs</a></li> -->
              <li class="divider"></li>
              <li class="dropdown-header">For sequence data:</li>
              <!-- <li class="divider"></li>   -->
              <li><a href="SummarySequenceData.html">Summary statistics</a></li>
              <li><a href="PopDiffSequenceData.html">Population differentiation</a></li>
          </ol>
        </li>
        <!-- Appendices dropdown -->
        <li class="dropdown">
          <a href="appendices" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Package Developers <span class="caret"></span></a>
          <ul class = "dropdown-menu" role = "menu">
           <li><a href = "DEVELOPER_INTRO.html">Introduction</a></li>
           <li><a href = "PACKAGES.html">Population genetic packages in R</a></li>  
           <li><a href = "DATAFORMATS.html">Population genetic data types in R</a></li>
           <li><a href = "PACKAGE_TUTORIALS.html">Developing R packages</a></li>
           <li class="divider"></li>  
           <li class="dropdown-header">Development Resources (external):</li>
           <li><a href = "http://r-pkgs.had.co.nz/">Hadley Wickham's Guide For Creating R Packages</a></li>
           <li><a href = "https://github.com/jtleek/rpackages">Jeff Leek's R Package Development</a></li>
           <li><a href = "http://adv-r.had.co.nz/S4.html">S4 Classes in detail</a></li>
           <li><a href = "http://adv-r.had.co.nz/S3.html">S3 Classes in detail</a></li>
           <li><a href = "http://cran.r-project.org/doc/contrib/Genolini-S4tutorialV0-5en.pdf">A not so short introduction to S4</a></li>
          </ul>
        </li>
        <!-- Author button -->
        <li class="dropdown">
          <a href="about" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Contribute! <span class="caret"></span></a>
          <ul class = "dropdown-menu" role = "menu">
           <li><a href = "CONTRIBUTING.html">Contribution Guidelines</a></li>  
           <li><a href = "CONTRIBUTING_WITH_GIT2R.html">Contributing with git2r</a></li> 
           <li class="divider"></li>  
           <li><a href="https://github.com/NESCent/popgenInfo">Source repository</a></li>  
		  </ul>
        </li>
        <li class = "button">
          <a href="LinksToResources.html">Useful Links</a>
        </li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<!-- <div class="container">
<div class="row">
<div class="span10 offset1"> -->

<div id="header">
<h1 class="title">Population Differentiation for Sequence Data</h1>
<h4 class="author"><em>Margarita M. López-Uribe</em></h4>
<h4 class="date"><em>March 19, 2015</em></h4>
</div>

<div id="TOC">
<ul>
<li><a href="#introduction">Introduction</a></li>
<li><a href="#resourcespackages">Resources/Packages</a></li>
<li><a href="#import-data">Import data</a></li>
<li><a href="#pairwise-population-differentiation">Pairwise population differentiation</a></li>
<li><a href="#amova-analysis-of-molecular-variance">AMOVA (Analysis of Molecular Variance)</a></li>
<li><a href="#conclusions">Conclusions</a><ul>
<li><a href="#whats-next">What’s next</a></li>
</ul></li>
<li><a href="#contributors">Contributors</a></li>
<li><a href="#session-information">Session Information</a></li>
</ul>
</div>

<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>Now, we will discuss in more detail how to assess population genetic structure from sequence data. Assuming that you have a priori information about how the individuals are grouped in subpopulations, one can do the following analyses: 1) quantify pairwise subpopulation structure and their significance, 2) test for hierarchical structure among groups of subpopulations, and 3) use population clustering algorithms to corroborate the a priori grouping hypothesis. We will go into each of these analyses in this vignette.</p>
</div>
<div id="resourcespackages" class="section level1">
<h1>Resources/Packages</h1>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(apex)
<span class="kw">library</span>(adegenet)
<span class="kw">library</span>(ape)
<span class="kw">library</span>(pegas)
<span class="kw">library</span>(hierfstat)
<span class="kw">library</span>(mmod)
<span class="kw">library</span>(genetics)
<span class="kw">library</span>(poppr)</code></pre>
</div>
<div id="import-data" class="section level1">
<h1>Import data</h1>
<p>We will first upload FASTA files for each gene independently, and then we will create a multidna object that concatenates both genes (allData1). Finally, this multidna object (allData2) will be converted into a genind object (allData3), which will be used for downstream analyses.</p>
<pre class="sourceCode r"><code class="sourceCode r">myData1&lt;-<span class="st"> </span><span class="kw">read.FASTA</span>(<span class="st">&quot;Ebom_mt.fas&quot;</span>) 
tmp1&lt;-<span class="kw">as.matrix</span>(myData1)
tmp1
myData2&lt;-<span class="st"> </span><span class="kw">read.FASTA</span>(<span class="st">&quot;Ebom_CAD.fas&quot;</span>) 
tmp2&lt;-<span class="kw">as.matrix</span>(myData2)
tmp2
allData1&lt;-<span class="kw">list</span>(<span class="dt">mt=</span>tmp1[,<span class="dv">1</span>:<span class="dv">1271</span>], <span class="dt">CAD=</span>tmp2[,<span class="dv">1</span>:<span class="dv">673</span>])
tmp3&lt;-<span class="kw">as.matrix</span>(allData1)
allData2&lt;-<span class="kw">new</span>(<span class="st">&quot;multidna&quot;</span>,allData1)
<span class="kw">plot</span>(allData2, <span class="dt">cex=</span><span class="fl">0.2</span>)
allData3&lt;-<span class="kw">multidna2genind</span>(allData2,<span class="dt">mlst=</span><span class="ot">TRUE</span>)
<span class="kw">pop</span>(allData3)&lt;-(<span class="kw">rep</span>(<span class="kw">c</span>(<span class="st">&quot;CA&quot;</span>,<span class="st">&quot;Ch&quot;</span>,<span class="st">&quot;Am&quot;</span>,<span class="st">&quot;AF&quot;</span>), <span class="dt">each=</span><span class="dv">10</span>))
allData3</code></pre>
<pre class="sourceCode r"><code class="sourceCode r">myData1&lt;-<span class="st"> </span><span class="kw">read.FASTA</span>(<span class="st">&quot;../data/Ebom_mt.fas&quot;</span>) <span class="co"># this creates a DNAbin object</span>
tmp1&lt;-<span class="kw">as.matrix</span>(myData1) <span class="co"># converts the list of sequences into a matrix</span>
tmp1</code></pre>
<pre><code>## 40 DNA sequences in binary format stored in a matrix.
## 
## All sequences of same length: 1271 
## 
## Labels: EbomML016_CRIC EbomML088_CRIC EbomML178_CRIC EbomML202_CRIS EbomML312_CRIC EbomML313_CRIC ...
## 
## Base composition:
##     a     c     g     t 
## 0.342 0.097 0.105 0.456</code></pre>
<pre class="sourceCode r"><code class="sourceCode r">myData2&lt;-<span class="st"> </span><span class="kw">read.FASTA</span>(<span class="st">&quot;../data/Ebom_CAD.fas&quot;</span>) <span class="co"># this creates a DNAbin object</span>
tmp2&lt;-<span class="kw">as.matrix</span>(myData2) <span class="co"># converts the list of sequences into a matrix</span>
tmp2</code></pre>
<pre><code>## 40 DNA sequences in binary format stored in a matrix.
## 
## All sequences of same length: 673 
## 
## Labels: EbomML016_CRIC EbomML088_CRIC EbomML178_CRIC EbomML202_CRIS EbomML312_CRIC EbomML313_CRIC ...
## 
## Base composition:
##     a     c     g     t 
## 0.272 0.226 0.241 0.262</code></pre>
<pre class="sourceCode r"><code class="sourceCode r">allData1&lt;-<span class="kw">list</span>(<span class="dt">mt=</span>tmp1[,<span class="dv">1</span>:<span class="dv">1271</span>], <span class="dt">CAD=</span>tmp2[,<span class="dv">1</span>:<span class="dv">673</span>])
tmp3&lt;-<span class="kw">as.matrix</span>(allData1)
allData2&lt;-<span class="kw">new</span>(<span class="st">&quot;multidna&quot;</span>,allData1)
<span class="kw">plot</span>(allData2, <span class="dt">cex=</span><span class="fl">0.2</span>)</code></pre>
<p><img src="PopDiffSequenceData_files/figure-html/data_import_FASTA_real-1.png" title="" alt="" width="960" /></p>
<pre class="sourceCode r"><code class="sourceCode r">allData3&lt;-<span class="kw">multidna2genind</span>(allData2,<span class="dt">mlst=</span><span class="ot">TRUE</span>)
<span class="kw">pop</span>(allData3)&lt;-(<span class="kw">rep</span>(<span class="kw">c</span>(<span class="st">&quot;CA&quot;</span>,<span class="st">&quot;Ch&quot;</span>,<span class="st">&quot;Am&quot;</span>,<span class="st">&quot;AF&quot;</span>), <span class="dt">each=</span><span class="dv">10</span>))
allData3</code></pre>
<pre><code>## /// GENIND OBJECT /////////
## 
##  // 40 individuals; 2 loci; 36 alleles; size: 51 Kb
## 
##  // Basic content
##    @tab:  40 x 36 matrix of allele counts
##    @loc.n.all: number of alleles per locus (range: 17-19)
##    @loc.fac: locus factor for the 36 columns of @tab
##    @all.names: list of allele names for each locus
##    @ploidy: ploidy of each individual  (range: 1-1)
##    @type:  codom
##    @call: df2genind(X = xdfnum, ind.names = x@labels, ploidy = 1)
## 
##  // Optional content
##    @pop: population of each individual (group size range: 10-10)</code></pre>
</div>
<div id="pairwise-population-differentiation" class="section level1">
<h1>Pairwise population differentiation</h1>
<div id="overall-f-statistics" class="section level3">
<h3>Overall F-statistics</h3>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">diff_stats</span>(allData3) <span class="co"># this function calculates overall Nei&#39;s Gst, Hedrick&#39;s Gst and  of the dataset</span></code></pre>
<pre><code>## $per.locus
##            Hs        Ht       Gst Gprime_st         D
## mt  0.7421053 0.9117763 0.1860885 0.9058976 0.8772109
## CAD 0.7263158 0.9190789 0.2097351 0.9550196 0.9391026
## 
## $global
##        Hs        Ht   Gst_est Gprime_st     D_het    D_mean 
## 0.7342105 0.9154276 0.1979590 0.9315893 0.9090759 0.9071022</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">Phi_st_Meirmans</span>(allData3) <span class="co"># this function calculates overall PhiST, the Fst analog for DNA sequence data</span></code></pre>
<pre><code>## $per.locus
##        mt       CAD 
## 0.8817688 0.9443099 
## 
## $global
## [1] 0.8938182</code></pre>
</div>
<div id="pairwise-fst" class="section level3">
<h3>Pairwise Fst</h3>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">pairwise_Gst_Nei</span>(allData3, <span class="dt">linearized=</span><span class="ot">FALSE</span>) <span class="co">#Calculates pairwise Gst. If linearized =TRUE, it calculates 1/(1-Gst)  </span></code></pre>
<pre><code>##            CA         Ch         Am
## Ch 0.08227848                      
## Am 0.12513019 0.15853659           
## AF 0.14803625 0.18012422 0.15297092</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">pairwise_Gst_Hedrick</span>(allData3, <span class="dt">linearized=</span><span class="ot">FALSE</span>)<span class="co">#Calculates pairwise Gst. If linearized =TRUE, it calculates 1/(1-Gst&#39;)  </span></code></pre>
<pre><code>##           CA        Ch        Am
## Ch 0.6419753                    
## Am 0.9828211 1.0000000          
## AF 1.0000000 1.0000000 0.9002976</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">pairwise_D</span>(allData3, <span class="dt">linearized=</span><span class="ot">FALSE</span>, <span class="dt">hsht_mean=</span><span class="st">&quot;harmonic&quot;</span>) <span class="co">#Calculates pairwise Gst. If linearized =TRUE, it calculates 1/(1-D)  </span></code></pre>
<pre><code>##           CA        Ch        Am
## Ch 0.5636175                    
## Am 0.9780685 1.0000000          
## AF 1.0007468 1.0012306 0.8632757</code></pre>
</div>
<div id="testing-for-significance" class="section level3">
<h3>Testing for significance</h3>
<p>To estimate if populations are significantly different, we will generate 100 replicates of the dataset using the function chao_bootstrap(). Then, summary statistics (meand and 95% CI) will be calculated for each of the different parameters of population differentiation.</p>
<pre class="sourceCode r"><code class="sourceCode r">bs&lt;-<span class="kw">chao_bootstrap</span>(allData3, <span class="dt">nreps=</span><span class="dv">100</span>)
<span class="kw">summarise_bootstrap</span>(bs, Gst_Nei) <span class="co">#for Nei&#39;s Gst</span></code></pre>
<pre><code>## 
## Estimates for each locus
## Locus    Mean     95% CI
## mt   1.6594  (1.146-2.5155)
## CAD  1.6862  (1.1299-2.496)
## 
## Global Estimate based on average heterozygosity
## 1.66 (1.2326-2.1705)</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">summarise_bootstrap</span>(bs, Gst_Hedrick) <span class="co">#for Hedrick&#39;s Gst</span></code></pre>
<pre><code>## 
## Estimates for each locus
## Locus    Mean     95% CI
## mt   1.035   (1.0091-1.0886)
## CAD  1.0205  (1.0029-1.0604)
## 
## Global Estimate based on average heterozygosity
## 1.0279   (1.013-1.0709)</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">summarise_bootstrap</span>(bs, D_Jost) <span class="co">#for Jost&#39;s D</span></code></pre>
<pre><code>## 
## Estimates for each locus
## Locus    Mean     95% CI
## mt   0.9137  (0.8535-0.9577)
## CAD  0.9496  (0.887-0.9898)
## 
## Global Estimate based on average heterozygosity
## 0.9322   (0.89-0.9657)
## 
## Global Estimate based on harmonic mean of statistic
## 0.9309   (0.8887-0.9655)</code></pre>
</div>
</div>
<div id="amova-analysis-of-molecular-variance" class="section level1">
<h1>AMOVA (Analysis of Molecular Variance)</h1>
<p>Analysis of Molecular Variance (AMOVA) is a method for estimating population differentiation from molecular data taking into account the mutational distance between alleles. Unlike Fst, which quantifies genetic differentiation based on allele frequencies, AMOVA treats molecular data as vectors and estimates Euclidean distances between alleles. Furthermore, it is possible to test hypotheses about differentiation by grouping subpopulations in a hierarchical structure (Ref).</p>
<pre class="sourceCode r"><code class="sourceCode r">allData_dist&lt;-<span class="kw">dist.multidna</span>(allData2,<span class="dt">pool=</span><span class="ot">TRUE</span>)
<span class="kw">strata</span>(allData3)&lt;-<span class="kw">data.frame</span>(<span class="dt">regions=</span><span class="kw">c</span>(<span class="kw">rep</span>(<span class="st">&quot;West&quot;</span>, <span class="dv">20</span>), <span class="kw">rep</span>(<span class="st">&quot;East&quot;</span>, <span class="dv">20</span>)), <span class="dt">populations=</span><span class="kw">c</span>(<span class="kw">rep</span>(<span class="st">&quot;CA&quot;</span>, <span class="dv">10</span>), <span class="kw">rep</span>(<span class="st">&quot;Ch&quot;</span>, <span class="dv">10</span>),<span class="kw">rep</span>(<span class="st">&quot;Am&quot;</span>, <span class="dv">10</span>), <span class="kw">rep</span>(<span class="st">&quot;AF&quot;</span>, <span class="dv">10</span>)))
<span class="kw">amova</span>(allData_dist ~<span class="st"> </span>populations, <span class="dt">data=</span><span class="kw">strata</span>(allData3), <span class="dt">nperm =</span> <span class="dv">100</span>)</code></pre>
<pre><code>## 
##  Analysis of Molecular Variance
## 
## Call: amova(formula = allData_dist ~ populations, data = strata(allData3), 
##     nperm = 100)
## 
##                      SSD          MSD df
## populations 0.0009468508 3.156169e-04  3
## Error       0.0005930812 1.647448e-05 36
## Total       0.0015399320 3.948544e-05 39
## 
## Variance components:
##                 sigma2 P.value
## populations 2.9914e-05       0
## Error       1.6474e-05        
## 
## Variance coefficients:
##  a 
## 10</code></pre>
</div>
<div id="conclusions" class="section level1">
<h1>Conclusions</h1>
<div id="whats-next" class="section level2">
<h2>What’s next</h2>
<p>Information on further analysis that could be done, other workflows <a href="DifferentiationSNP.html">can be linked as well</a> (note the HTML as opposed to Rmd link).</p>
</div>
</div>
<div id="contributors" class="section level1">
<h1>Contributors</h1>
<ul>
<li>Author 1</li>
<li>Author 2</li>
<li>Contributor 1 (role)</li>
</ul>
</div>
<div id="session-information" class="section level1">
<h1>Session Information</h1>
<p>This shows us useful information for reproducibility. Of particular importance are the versions of R and the packages used to create this workflow. It is considered good practice to record this information with every analysis.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">width =</span> <span class="dv">100</span>)
devtools::<span class="kw">session_info</span>()</code></pre>
<pre><code>## Session info ---------------------------------------------------------------------------------------</code></pre>
<pre><code>##  setting  value                                 
##  version  R version 3.2.3 RC (2015-12-03 r69731)
##  system   x86_64, linux-gnu                     
##  ui       X11                                   
##  language (EN)                                  
##  collate  en_US.UTF-8                           
##  tz       &lt;NA&gt;                                  
##  date     2015-12-24</code></pre>
<pre><code>## Packages -------------------------------------------------------------------------------------------</code></pre>
<pre><code>##  package      * version date       source        
##  ade4         * 1.7-3   2015-11-22 CRAN (R 3.2.3)
##  adegenet     * 2.0.0   2015-07-07 CRAN (R 3.2.3)
##  ape          * 3.4     2015-11-29 CRAN (R 3.2.3)
##  apex         * 1.0.1   2015-09-07 CRAN (R 3.2.3)
##  assertthat     0.1     2013-12-06 CRAN (R 3.2.3)
##  BiocGenerics   0.16.1  2015-12-23 Bioconductor  
##  Biostrings     2.38.2  2015-12-23 Bioconductor  
##  boot           1.3-17  2015-06-29 CRAN (R 3.2.1)
##  cluster        2.0.3   2015-07-21 CRAN (R 3.2.1)
##  coda           0.18-1  2015-10-16 CRAN (R 3.2.3)
##  colorspace     1.2-6   2015-03-11 CRAN (R 3.2.3)
##  combinat     * 0.0-8   2012-10-29 CRAN (R 3.2.3)
##  DBI            0.3.1   2014-09-24 CRAN (R 3.2.3)
##  deldir         0.1-9   2015-03-09 CRAN (R 3.2.3)
##  devtools       1.9.1   2015-09-11 CRAN (R 3.2.3)
##  digest         0.6.8   2014-12-31 CRAN (R 3.2.3)
##  dplyr          0.4.3   2015-09-01 CRAN (R 3.2.3)
##  evaluate       0.8     2015-09-18 CRAN (R 3.2.3)
##  formatR        1.2.1   2015-09-18 CRAN (R 3.2.3)
##  gdata        * 2.17.0  2015-07-04 CRAN (R 3.2.3)
##  genetics     * 1.3.8.1 2013-09-03 CRAN (R 3.2.3)
##  ggplot2        2.0.0   2015-12-18 CRAN (R 3.2.3)
##  gtable         0.1.2   2012-12-05 CRAN (R 3.2.3)
##  gtools       * 3.5.0   2015-05-29 CRAN (R 3.2.3)
##  hierfstat    * 0.04-22 2015-12-04 CRAN (R 3.2.3)
##  htmltools      0.2.6   2014-09-08 CRAN (R 3.2.3)
##  httpuv         1.3.3   2015-08-04 CRAN (R 3.2.3)
##  igraph         1.0.1   2015-06-26 CRAN (R 3.2.3)
##  IRanges        2.4.6   2015-12-23 Bioconductor  
##  knitr          1.11    2015-08-14 CRAN (R 3.2.3)
##  lattice        0.20-33 2015-07-14 CRAN (R 3.2.3)
##  LearnBayes     2.15    2014-05-29 CRAN (R 3.2.3)
##  magrittr       1.5     2014-11-22 CRAN (R 3.2.3)
##  MASS         * 7.3-45  2015-11-10 CRAN (R 3.2.3)
##  Matrix         1.2-3   2015-11-28 CRAN (R 3.2.2)
##  memoise        0.2.1   2014-04-22 CRAN (R 3.2.3)
##  mgcv           1.8-10  2015-12-12 CRAN (R 3.2.3)
##  mime           0.4     2015-09-03 CRAN (R 3.2.3)
##  mmod         * 1.3.1   2015-08-27 CRAN (R 3.2.3)
##  munsell        0.4.2   2013-07-11 CRAN (R 3.2.3)
##  mvtnorm      * 1.0-3   2015-07-22 CRAN (R 3.2.3)
##  nlme           3.1-122 2015-08-19 CRAN (R 3.2.3)
##  nnls           1.4     2012-03-19 CRAN (R 3.2.3)
##  pegas        * 0.8-2   2015-09-25 CRAN (R 3.2.3)
##  permute        0.8-4   2015-05-19 CRAN (R 3.2.3)
##  phangorn     * 2.0.1   2015-12-15 CRAN (R 3.2.3)
##  plyr           1.8.3   2015-06-12 CRAN (R 3.2.3)
##  poppr        * 2.1.0   2015-12-01 CRAN (R 3.2.3)
##  quadprog       1.5-5   2013-04-17 CRAN (R 3.2.3)
##  R6             2.1.1   2015-08-19 CRAN (R 3.2.3)
##  Rcpp           0.12.2  2015-11-15 CRAN (R 3.2.3)
##  reshape2       1.4.1   2014-12-06 CRAN (R 3.2.3)
##  rmarkdown      0.9     2015-12-22 CRAN (R 3.2.3)
##  S4Vectors      0.8.5   2015-12-23 Bioconductor  
##  scales         0.3.0   2015-08-25 CRAN (R 3.2.3)
##  seqinr         3.1-3   2014-12-17 CRAN (R 3.2.3)
##  shiny          0.12.2  2015-08-05 CRAN (R 3.2.3)
##  sp             1.2-1   2015-10-18 CRAN (R 3.2.3)
##  spdep          0.5-92  2015-12-22 CRAN (R 3.2.3)
##  stringi        1.0-1   2015-10-22 CRAN (R 3.2.3)
##  stringr        1.0.0   2015-04-30 CRAN (R 3.2.3)
##  vegan          2.3-2   2015-11-19 CRAN (R 3.2.3)
##  xtable         1.8-0   2015-11-02 CRAN (R 3.2.3)
##  XVector        0.10.0  2015-12-23 Bioconductor  
##  yaml           2.1.13  2014-06-12 CRAN (R 3.2.3)
##  zlibbioc       1.16.0  2015-12-23 Bioconductor</code></pre>
</div>

<!-- </div>
</div>
</div>
 -->

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
