<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Population Differentiation for Sequence Data</title>

<script src="libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.5/css/spacelab.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="libs/navigation-1.1/tabsets.js"></script>
<!---

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
  padding-right: 50px;
  padding-left: -50px;
}
@media (max-width: 979px) {
  body {
    padding-top: 0;
  }
}

 offset scroll position for anchor links (for fixed navbar)  
@media (min-width: 980px) {
  .section h2 {
    padding-top: 52px;
    margin-top: -52px;
  }
  .section h3 {
    padding-top: 52px;
    margin-top: -52px;
  }
}


/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

</style>
-->



<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}
/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}
/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}
</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #f8f8f8; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
pre, code { background-color: #f8f8f8; }
code > span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code > span.dt { color: #204a87; } /* DataType */
code > span.dv { color: #0000cf; } /* DecVal */
code > span.bn { color: #0000cf; } /* BaseN */
code > span.fl { color: #0000cf; } /* Float */
code > span.ch { color: #4e9a06; } /* Char */
code > span.st { color: #4e9a06; } /* String */
code > span.co { color: #8f5902; font-style: italic; } /* Comment */
code > span.ot { color: #8f5902; } /* Other */
code > span.al { color: #ef2929; } /* Alert */
code > span.fu { color: #000000; } /* Function */
code > span.er { color: #a40000; font-weight: bold; } /* Error */
code > span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #000000; } /* Constant */
code > span.sc { color: #000000; } /* SpecialChar */
code > span.vs { color: #4e9a06; } /* VerbatimString */
code > span.ss { color: #4e9a06; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #000000; } /* Variable */
code > span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code > span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code > span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code > span.ex { } /* Extension */
code > span.at { color: #c4a000; } /* Attribute */
code > span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code > span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <!-- HOME -->
      <a class="navbar-brand" href="index.html"><b>Population Genetics in R</b></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <!-- Table of contents -->
        <li class="dropdown">
          <a href="chapters" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Users <span class="caret"></span></a>
          <ol class="dropdown-menu">
              <li><a href="GETSTARTED.html">Getting Started</a></li>
              <li class="divider"></li>
              <li class="dropdown-header">For microsatellite data:</li>
              <li><a href="startMicrosatellite.html">Basic statistics</a></li>
              <li><a href="2015-12-15-microsatellite-differentiation.html">Population differentiation</a></li>
              <li><a href="LMM-Genetic-Diversity.html">Comparing genetic diversity using mixed models</a></li>
              <li class="divider"></li>
              <li class="dropdown-header">For SNP data:</li>
              <!-- <li class="divider"></li> -->
              <li><a href="StartSNP.html">Basic statistics</a></li>
              <li><a href="DifferentiationSNP.html">Population differentiation</a></li>
              <li><a href="2015-05-18-Dist-SNP.html">Genetic distance (individual-based)</a></li>
              <li><a href="2016-01-26-SNP-selection.html">Signals of selection</a></li>
			  <li><a href="2016-12-13_MEM_outlier.html">Spatial Outlier Detection</a></li>
              <!-- <li><a href="SNP.html">SNPs</a></li> -->
              <li class="divider"></li>
              <li class="dropdown-header">For sequence data:</li>
              <!-- <li class="divider"></li>   -->
              <li><a href="PopDiffSequenceData.html">Population differentiation</a></li>
          </ol>
        </li>
        <!-- Appendices dropdown -->
        <li class="dropdown">
          <a href="appendices" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Package Developers <span class="caret"></span></a>
          <ul class = "dropdown-menu" role = "menu">
           <li><a href = "PACKAGES.html">Population genetic packages in R</a></li>  
           <li><a href = "DATAFORMATS.html">Population genetic data types in R</a></li>
           <li><a href = "PACKAGE_TUTORIALS.html">Tutorials for developing R packages</a></li>
           <li class="divider"></li>  
           <li class="dropdown-header">Development Resources (external):</li>
           <li><a href = "http://r-pkgs.had.co.nz/">Hadley Wickham's guide for creating R packages</a></li>
           <li><a href = "https://github.com/jtleek/rpackages">Jeff Leek's R package development</a></li>
           <li><a href = "http://adv-r.had.co.nz/S4.html">S4 Classes in detail</a></li>
           <li><a href = "http://adv-r.had.co.nz/S3.html">S3 Classes in detail</a></li>
           <li><a href = "http://cran.r-project.org/doc/contrib/Genolini-S4tutorialV0-5en.pdf">A not so short introduction to S4</a></li>
          </ul>
        </li>
        <!-- Author button -->
        <li class="dropdown">
          <a href="about" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Contribute! <span class="caret"></span></a>
          <ul class = "dropdown-menu" role = "menu">
           <li><a href = "CONTRIBUTING.html">Contribution guidelines</a></li>  
           <li><a href = "CONTRIBUTING_WITH_GIT2R.html">Contributing with git2r</a></li> 
           <li class="divider"></li>  
           <li><a href="https://github.com/NESCent/popgenInfo">Source repository</a></li>  
		  </ul>
        </li>
        <li class = "button">
          <a href="LinksToResources.html">Useful Links</a>
        </li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<!-- <div class="container">
<div class="row">
<div class="span10 offset1"> -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Population Differentiation for Sequence Data</h1>

</div>

<div id="TOC">
<ul>
<li><a href="#introduction">Introduction</a></li>
<li><a href="#resources">Resources</a><ul>
<li><a href="#data">Data</a></li>
<li><a href="#packages">Packages</a></li>
</ul></li>
<li><a href="#importing-data">Importing data</a></li>
<li><a href="#pairwise-population-differentiation">Pairwise population differentiation</a><ul>
<li><a href="#overall-f-statistics">Overall F-statistics</a></li>
<li><a href="#pairwise-fst">Pairwise Fst</a></li>
<li><a href="#testing-for-significance">Testing for significance</a></li>
</ul></li>
<li><a href="#amova-analysis-of-molecular-variance">AMOVA (Analysis of Molecular Variance)</a></li>
<li><a href="#conclusions">Conclusions</a><ul>
<li><a href="#whats-next">What’s next</a></li>
</ul></li>
<li><a href="#contributors">Contributors</a></li>
<li><a href="#references">References</a></li>
<li><a href="#session-information">Session Information</a></li>
</ul>
</div>

<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>Now, we will discuss in more detail how to assess population genetic structure from sequence data. Assuming that you have a priori information about how the individuals are grouped in subpopulations, one can do the following analyses: 1) quantify pairwise subpopulation structure and their significance, 2) test for hierarchical structure among groups of subpopulations, and 3) use population clustering algorithms to corroborate the a priori grouping hypothesis. We will go into each of these analyses in this vignette.</p>
</div>
<div id="resources" class="section level1">
<h1>Resources</h1>
<div id="data" class="section level2">
<h2>Data</h2>
<p>Data used for the following analyses come from one mitochondrial gene (Cytochrome Oxidase I = <a href="https://github.com/nescent/popgenInfo/blob/master/data/Ebom_mt.fas">‘Ebom_mt.fas’</a>) and one nuclear gene (Calcium/Calmodulin-Dependent Protein Kinase II = <a href="https://github.com/nescent/popgenInfo/blob/master/data/Ebom_CAD.fas">‘Ebom_CAD.fas’</a>) of the orchid bee <em>Eulaema bombiformis</em>.</p>
</div>
<div id="packages" class="section level2">
<h2>Packages</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;apex&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;adegenet&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;pegas&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;mmod&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;poppr&quot;</span>)</code></pre></div>
</div>
</div>
<div id="importing-data" class="section level1">
<h1>Importing data</h1>
<p>We will first import FASTA files for all genes at once using the <em>apex</em> function <code>read.multiFASTA()</code>. This will allow us to read in multiple FASTA formatted files at once into a “multiDNA” object from the <em>apex</em> package. We can plot it to see what the two genes look like for all of our samples</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Creating DNAbin objects</span>
beeData &lt;-<span class="st"> </span><span class="kw">read.multiFASTA</span>(<span class="kw">c</span>(<span class="st">&quot;Ebom_mt.fas&quot;</span>, <span class="st">&quot;Ebom_CAD.fas&quot;</span>))
<span class="kw">plot</span>(beeData, <span class="dt">cex =</span> <span class="fl">0.2</span>)</code></pre></div>
<p><img src="PopDiffSequenceData_files/figure-html/data_import_FASTA_real-1.png" width="960" /></p>
<blockquote>
<p>Pro tip: You can list all the files ending in “.fas” in your folder by using <code>dir(pattern = &quot;fas&quot;, full.names = TRUE)</code></p>
</blockquote>
<p>Now, this “multidna” object (<code>beeData</code>) will be converted into a “genind” object (<code>beeData.gid</code>), which will be used for downstream analyses. One thing we need to be aware of is the fact that we can’t have any periods in our locus names. By default, the locus names are set to the file names, which both contain “.fas” in the title. We can fix this by using the text-replacing function <code>gsub()</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">getLocusNames</span>(beeData)</code></pre></div>
<pre><code>## [1] &quot;Ebom_mt.fas&quot;  &quot;Ebom_CAD.fas&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">(<span class="kw">setLocusNames</span>(beeData) &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;.fas&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="kw">getLocusNames</span>(beeData)))</code></pre></div>
<pre><code>## [1] &quot;Ebom_mt&quot;  &quot;Ebom_CAD&quot;</code></pre>
<p>Now we can create our “genind” object.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Creating genind object by multilocus sequence types</span>
beeData.gid &lt;-<span class="st"> </span><span class="kw">multidna2genind</span>(beeData, <span class="dt">mlst =</span> <span class="ot">TRUE</span>)
beeData.gid</code></pre></div>
<pre><code>## /// GENIND OBJECT /////////
## 
##  // 40 individuals; 2 loci; 36 alleles; size: 50.6 Kb
## 
##  // Basic content
##    @tab:  40 x 36 matrix of allele counts
##    @loc.n.all: number of alleles per locus (range: 17-19)
##    @loc.fac: locus factor for the 36 columns of @tab
##    @all.names: list of allele names for each locus
##    @ploidy: ploidy of each individual  (range: 1-1)
##    @type:  codom
##    @call: df2genind(X = xdfnum, ind.names = x@labels, ploidy = 1)
## 
##  // Optional content
##    - empty -</code></pre>
<p>We also want to set the population strata.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">my_strata &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">regions =</span> <span class="kw">rep</span>(<span class="kw">c</span>(<span class="st">&quot;West&quot;</span>, <span class="st">&quot;East&quot;</span>), <span class="dt">each =</span> <span class="dv">20</span>), 
                        <span class="dt">populations =</span> <span class="kw">rep</span>(<span class="kw">c</span>(<span class="st">&quot;CA&quot;</span>, <span class="st">&quot;Ch&quot;</span>, <span class="st">&quot;Am&quot;</span>, <span class="st">&quot;AF&quot;</span>), <span class="dt">each =</span> <span class="dv">10</span>))
<span class="kw">strata</span>(beeData.gid) &lt;-<span class="st"> </span>my_strata
<span class="kw">setPop</span>(beeData.gid) &lt;-<span class="st"> </span><span class="er">~</span>populations
beeData.gid</code></pre></div>
<pre><code>## /// GENIND OBJECT /////////
## 
##  // 40 individuals; 2 loci; 36 alleles; size: 53.4 Kb
## 
##  // Basic content
##    @tab:  40 x 36 matrix of allele counts
##    @loc.n.all: number of alleles per locus (range: 17-19)
##    @loc.fac: locus factor for the 36 columns of @tab
##    @all.names: list of allele names for each locus
##    @ploidy: ploidy of each individual  (range: 1-1)
##    @type:  codom
##    @call: df2genind(X = xdfnum, ind.names = x@labels, ploidy = 1)
## 
##  // Optional content
##    @pop: population of each individual (group size range: 10-10)
##    @strata: a data frame with 2 columns ( regions, populations )</code></pre>
</div>
<div id="pairwise-population-differentiation" class="section level1">
<h1>Pairwise population differentiation</h1>
<div id="overall-f-statistics" class="section level2">
<h2>Overall F-statistics</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">diff_stats</span>(beeData.gid) <span class="co"># this function calculates overall Nei&#39;s Gst, Hedrick&#39;s Gst and  of the dataset</span></code></pre></div>
<pre><code>## $per.locus
##                 Hs        Ht       Gst Gprime_st         D
## Ebom_mt  0.7421053 0.9117763 0.1860885 0.9058976 0.8772109
## Ebom_CAD 0.7263158 0.9190789 0.2097351 0.9550196 0.9391026
## 
## $global
##        Hs        Ht   Gst_est Gprime_st     D_het    D_mean 
## 0.7342105 0.9154276 0.1979590 0.9315893 0.9090759 0.9071022</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">Phi_st_Meirmans</span>(beeData.gid) <span class="co"># this function calculates overall PhiST, the Fst analog for DNA sequence data</span></code></pre></div>
<pre><code>## $per.locus
##   Ebom_mt  Ebom_CAD 
## 0.8817688 0.9443099 
## 
## $global
## [1] 0.8938182</code></pre>
</div>
<div id="pairwise-fst" class="section level2">
<h2>Pairwise Fst</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">pairwise_Gst_Nei</span>(beeData.gid, <span class="dt">linearized =</span> <span class="ot">FALSE</span>) <span class="co"># Calculates pairwise Gst. If linearized = TRUE, it calculates 1/(1- Gst)  </span></code></pre></div>
<pre><code>##            CA         Ch         Am
## Ch 0.08227848                      
## Am 0.12513019 0.15853659           
## AF 0.14803625 0.18012422 0.15297092</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">pairwise_Gst_Hedrick</span>(beeData.gid, <span class="dt">linearized =</span> <span class="ot">FALSE</span>)<span class="co"># Calculates pairwise Gst. If linearized = TRUE, it calculates 1/(1- Gst&#39;)  </span></code></pre></div>
<pre><code>##           CA        Ch        Am
## Ch 0.6419753                    
## Am 0.9828211 1.0000000          
## AF 1.0000000 1.0000000 0.9002976</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">pairwise_D</span>(beeData.gid, <span class="dt">linearized =</span> <span class="ot">FALSE</span>, <span class="dt">hsht_mean =</span> <span class="st">&quot;harmonic&quot;</span>) <span class="co"># Calculates pairwise Gst. If linearized = TRUE, it calculates 1/(1- D)  </span></code></pre></div>
<pre><code>##           CA        Ch        Am
## Ch 0.5636175                    
## Am 0.9780685 1.0000000          
## AF 1.0007468 1.0012306 0.8632757</code></pre>
</div>
<div id="testing-for-significance" class="section level2">
<h2>Testing for significance</h2>
<p>To estimate if populations are significantly different, we will generate 100 replicates of the dataset using the function <code>chao_bootstrap()</code>. Then, summary statistics (mean and 95% CI) will be calculated for each of the different parameters of population differentiation.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">bs &lt;-<span class="st"> </span><span class="kw">chao_bootstrap</span>(beeData.gid, <span class="dt">nreps =</span> <span class="dv">100</span>)
<span class="kw">summarise_bootstrap</span>(bs, Gst_Nei)     <span class="co"># for Nei&#39;s Gst</span></code></pre></div>
<pre><code>## 
## Estimates for each locus
## Locus    Mean     95% CI
## Ebom_mt  0.1861  (0.085-0.288)
## Ebom_CAD 0.2097  (0.080-0.340)
## 
## Global Estimate based on average heterozygosity
## 0.1980   (0.123-0.273)</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">summarise_bootstrap</span>(bs, Gst_Hedrick) <span class="co"># for Hedrick&#39;s Gst</span></code></pre></div>
<pre><code>## 
## Estimates for each locus
## Locus    Mean     95% CI
## Ebom_mt  0.9059  (0.837-0.975)
## Ebom_CAD 0.9550  (0.909-1.001)
## 
## Global Estimate based on average heterozygosity
## 0.9316   (0.887-0.976)</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">summarise_bootstrap</span>(bs, D_Jost)      <span class="co"># for Jost&#39;s D</span></code></pre></div>
<pre><code>## 
## Estimates for each locus
## Locus    Mean     95% CI
## Ebom_mt  0.8772  (0.784-0.970)
## Ebom_CAD 0.9391  (0.875-1.003)
## 
## Global Estimate based on average heterozygosity
## 0.9091   (0.848-0.970)
## 
## Global Estimate based on harmonic mean of statistic
## 0.9071   (0.846-0.968)</code></pre>
</div>
</div>
<div id="amova-analysis-of-molecular-variance" class="section level1">
<h1>AMOVA (Analysis of Molecular Variance)</h1>
<p>Analysis of Molecular Variance (AMOVA) is a method for estimating population differentiation from molecular data taking into account the mutational distance between alleles. Unlike <span class="math inline">\(F_{st}\)</span>, which quantifies genetic differentiation based on allele frequencies, AMOVA treats molecular data as vectors and estimates Euclidean distances between alleles. Furthermore, it is possible to test hypotheses about differentiation by grouping subpopulations in a hierarchical structure (Excoffier et al., 1992).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">beeData_dist &lt;-<span class="st"> </span><span class="kw">dist.multidna</span>(beeData, <span class="dt">pool =</span> <span class="ot">TRUE</span>)
<span class="kw">amova</span>(beeData_dist <span class="op">~</span><span class="st"> </span>populations, <span class="dt">data =</span> <span class="kw">strata</span>(beeData.gid), <span class="dt">nperm =</span> <span class="dv">100</span>)</code></pre></div>
<pre><code>## 
##  Analysis of Molecular Variance
## 
## Call: amova(formula = beeData_dist ~ populations, data = strata(beeData.gid), 
##     nperm = 100)
## 
##                      SSD          MSD df
## populations 0.0009468508 3.156169e-04  3
## Error       0.0005930812 1.647448e-05 36
## Total       0.0015399320 3.948544e-05 39
## 
## Variance components:
##                 sigma2 P.value
## populations 2.9914e-05       0
## Error       1.6474e-05        
## 
## Variance coefficients:
##  a 
## 10</code></pre>
</div>
<div id="conclusions" class="section level1">
<h1>Conclusions</h1>
<p>In this vignette, we learned how to estimate different parameters of overall and pairwise population differentiation (Nei’s Gst, Hedrick’s Gst, Jost D) from DNA sequence data. We also learned how to do a hierarchical analysis of population structure using AMOVA.</p>
<div id="whats-next" class="section level2">
<h2>What’s next</h2>
<p>With these exploratory analyses, we understand how genetic diversity is distributed among populations. One could now move to testing specific hypotheses than could explain the patterns of population structure found with the F-statistics.</p>
</div>
</div>
<div id="contributors" class="section level1">
<h1>Contributors</h1>
<ul>
<li>Margarita M. López- Uribe (Author)</li>
<li>Zhian N. Kamvar (edits)</li>
</ul>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<p>Excoffier L., Smouse PE., Quattro JM. 1992. Analysis of molecular variance inferred from metric distances among dNA haplotypes: Application to human mitochondrial dNA restriction data. Genetics 131:479–491. Available at: <a href="http://www.genetics.org/content/131/2/479.abstract" class="uri">http://www.genetics.org/content/131/2/479.abstract</a></p>
</div>
<div id="session-information" class="section level1">
<h1>Session Information</h1>
<p>This shows us useful information for reproducibility. Of particular importance are the versions of R and the packages used to create this workflow. It is considered good practice to record this information with every analysis.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">width =</span> <span class="dv">100</span>)
devtools<span class="op">::</span><span class="kw">session_info</span>()</code></pre></div>
<pre><code>## Session info --------------------------------------------------------------------------------------</code></pre>
<pre><code>##  setting  value                       
##  version  R version 3.4.2 (2017-09-28)
##  system   x86_64, linux-gnu           
##  ui       X11                         
##  language (EN)                        
##  collate  en_US.UTF-8                 
##  tz       UTC                         
##  date     2017-10-22</code></pre>
<pre><code>## Packages ------------------------------------------------------------------------------------------</code></pre>
<pre><code>##  package    * version date       source        
##  ade4       * 1.7-8   2017-08-09 CRAN (R 3.4.2)
##  adegenet   * 2.1.0   2017-10-12 CRAN (R 3.4.2)
##  ape        * 4.1     2017-02-14 CRAN (R 3.4.2)
##  apex       * 1.0.3   2017-09-23 CRAN (R 3.4.2)
##  assertthat   0.2.0   2017-04-11 CRAN (R 3.4.2)
##  backports    1.1.1   2017-09-25 CRAN (R 3.4.2)
##  base       * 3.4.2   2017-10-21 local         
##  bindr        0.1     2016-11-13 CRAN (R 3.4.2)
##  bindrcpp     0.2     2017-06-17 CRAN (R 3.4.2)
##  boot         1.3-20  2017-07-30 CRAN (R 3.4.2)
##  cluster      2.0.6   2017-03-16 CRAN (R 3.4.2)
##  coda         0.19-1  2016-12-08 CRAN (R 3.4.2)
##  colorspace   1.3-2   2016-12-14 CRAN (R 3.4.2)
##  compiler     3.4.2   2017-10-21 local         
##  datasets   * 3.4.2   2017-10-21 local         
##  deldir       0.1-14  2017-04-22 CRAN (R 3.4.2)
##  devtools     1.13.3  2017-08-02 CRAN (R 3.4.2)
##  digest       0.6.12  2017-01-27 CRAN (R 3.4.2)
##  dplyr        0.7.4   2017-09-28 CRAN (R 3.4.2)
##  evaluate     0.10.1  2017-06-24 CRAN (R 3.4.2)
##  expm         0.999-2 2017-03-29 CRAN (R 3.4.2)
##  fastmatch    1.1-0   2017-01-28 CRAN (R 3.4.2)
##  gdata        2.18.0  2017-06-06 CRAN (R 3.4.2)
##  ggplot2      2.2.1   2016-12-30 CRAN (R 3.4.2)
##  glue         1.1.1   2017-06-21 CRAN (R 3.4.2)
##  gmodels      2.16.2  2015-07-22 CRAN (R 3.4.2)
##  graphics   * 3.4.2   2017-10-21 local         
##  grDevices  * 3.4.2   2017-10-21 local         
##  grid         3.4.2   2017-10-21 local         
##  gtable       0.2.0   2016-02-26 CRAN (R 3.4.2)
##  gtools       3.5.0   2015-05-29 CRAN (R 3.4.2)
##  htmltools    0.3.6   2017-04-28 CRAN (R 3.4.2)
##  httpuv       1.3.5   2017-07-04 CRAN (R 3.4.2)
##  igraph       1.1.2   2017-07-21 CRAN (R 3.4.2)
##  knitr        1.17    2017-08-10 CRAN (R 3.4.2)
##  lattice      0.20-35 2017-03-25 CRAN (R 3.4.2)
##  lazyeval     0.2.0   2016-06-12 CRAN (R 3.4.2)
##  LearnBayes   2.15    2014-05-29 CRAN (R 3.4.2)
##  magrittr     1.5     2014-11-22 CRAN (R 3.4.2)
##  MASS         7.3-47  2017-04-21 CRAN (R 3.4.2)
##  Matrix       1.2-11  2017-08-16 CRAN (R 3.4.2)
##  memoise      1.1.0   2017-04-21 CRAN (R 3.4.2)
##  methods    * 3.4.2   2017-10-21 local         
##  mgcv         1.8-22  2017-09-19 CRAN (R 3.4.2)
##  mime         0.5     2016-07-07 CRAN (R 3.4.2)
##  mmod       * 1.3.3   2017-04-06 CRAN (R 3.4.2)
##  munsell      0.4.3   2016-02-13 CRAN (R 3.4.2)
##  nlme         3.1-131 2017-02-06 CRAN (R 3.4.2)
##  parallel     3.4.2   2017-10-21 local         
##  pegas      * 0.10    2017-05-03 CRAN (R 3.4.2)
##  permute      0.9-4   2016-09-09 CRAN (R 3.4.2)
##  phangorn   * 2.2.0   2017-04-03 CRAN (R 3.4.2)
##  pkgconfig    2.0.1   2017-03-21 CRAN (R 3.4.2)
##  plyr         1.8.4   2016-06-08 CRAN (R 3.4.2)
##  poppr      * 2.5.0   2017-09-11 CRAN (R 3.4.2)
##  quadprog     1.5-5   2013-04-17 CRAN (R 3.4.2)
##  R6           2.2.2   2017-06-17 CRAN (R 3.4.2)
##  Rcpp         0.12.13 2017-09-28 CRAN (R 3.4.2)
##  reshape2     1.4.2   2016-10-22 CRAN (R 3.4.2)
##  rlang        0.1.2   2017-08-09 CRAN (R 3.4.2)
##  rmarkdown    1.6     2017-06-15 CRAN (R 3.4.2)
##  rprojroot    1.2     2017-01-16 CRAN (R 3.4.2)
##  scales       0.5.0   2017-08-24 CRAN (R 3.4.2)
##  seqinr       3.4-5   2017-08-01 CRAN (R 3.4.2)
##  shiny        1.0.5   2017-08-23 CRAN (R 3.4.2)
##  sp           1.2-5   2017-06-29 CRAN (R 3.4.2)
##  spdep        0.6-15  2017-09-01 CRAN (R 3.4.2)
##  splines      3.4.2   2017-10-21 local         
##  stats      * 3.4.2   2017-10-21 local         
##  stringi      1.1.5   2017-04-07 CRAN (R 3.4.2)
##  stringr      1.2.0   2017-02-18 CRAN (R 3.4.2)
##  tibble       1.3.4   2017-08-22 CRAN (R 3.4.2)
##  tools        3.4.2   2017-10-21 local         
##  utils      * 3.4.2   2017-10-21 local         
##  vegan        2.4-4   2017-08-24 CRAN (R 3.4.2)
##  withr        2.0.0   2017-07-28 CRAN (R 3.4.2)
##  xtable       1.8-2   2016-02-05 CRAN (R 3.4.2)
##  yaml         2.1.14  2016-11-12 CRAN (R 3.4.2)</code></pre>
</div>

<!-- </div>
</div>
</div>
 -->



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
