<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Calculating Basic Population Genetic Statistics from SNP Data</title>

<script src="libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.5/css/spacelab.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="libs/navigation-1.1/tabsets.js"></script>
<!---

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
  padding-right: 50px;
  padding-left: -50px;
}
@media (max-width: 979px) {
  body {
    padding-top: 0;
  }
}

 offset scroll position for anchor links (for fixed navbar)  
@media (min-width: 980px) {
  .section h2 {
    padding-top: 52px;
    margin-top: -52px;
  }
  .section h3 {
    padding-top: 52px;
    margin-top: -52px;
  }
}


/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

</style>
-->



<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}
/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}
/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}
</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #f8f8f8; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
pre, code { background-color: #f8f8f8; }
code > span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code > span.dt { color: #204a87; } /* DataType */
code > span.dv { color: #0000cf; } /* DecVal */
code > span.bn { color: #0000cf; } /* BaseN */
code > span.fl { color: #0000cf; } /* Float */
code > span.ch { color: #4e9a06; } /* Char */
code > span.st { color: #4e9a06; } /* String */
code > span.co { color: #8f5902; font-style: italic; } /* Comment */
code > span.ot { color: #8f5902; } /* Other */
code > span.al { color: #ef2929; } /* Alert */
code > span.fu { color: #000000; } /* Function */
code > span.er { color: #a40000; font-weight: bold; } /* Error */
code > span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #000000; } /* Constant */
code > span.sc { color: #000000; } /* SpecialChar */
code > span.vs { color: #4e9a06; } /* VerbatimString */
code > span.ss { color: #4e9a06; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #000000; } /* Variable */
code > span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code > span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code > span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code > span.ex { } /* Extension */
code > span.at { color: #c4a000; } /* Attribute */
code > span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code > span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <!-- HOME -->
      <a class="navbar-brand" href="index.html"><b>Population Genetics in R</b></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <!-- Table of contents -->
        <li class="dropdown">
          <a href="chapters" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Users <span class="caret"></span></a>
          <ol class="dropdown-menu">
              <li><a href="GETSTARTED.html">Getting Started</a></li>
              <li class="divider"></li>
              <li class="dropdown-header">For microsatellite data:</li>
              <li><a href="startMicrosatellite.html">Basic statistics</a></li>
              <li><a href="2015-12-15-microsatellite-differentiation.html">Population differentiation</a></li>
              <li><a href="LMM-Genetic-Diversity.html">Comparing genetic diversity using mixed models</a></li>
              <li class="divider"></li>
              <li class="dropdown-header">For SNP data:</li>
              <!-- <li class="divider"></li> -->
              <li><a href="StartSNP.html">Basic statistics</a></li>
              <li><a href="DifferentiationSNP.html">Population differentiation</a></li>
              <li><a href="2015-05-18-Dist-SNP.html">Genetic distance (individual-based)</a></li>
              <li><a href="2016-01-26-SNP-selection.html">Signals of selection</a></li>
			  <li><a href="2016-12-13_MEM_outlier.html">Spatial outlier detection</a></li>
			  <li><a href="2018-03-27_RDA_GEA.html">Detecting multilocus adaptation using redundancy analysis</a></li>
              <!-- <li><a href="SNP.html">SNPs</a></li> -->
              <li class="divider"></li>
              <li class="dropdown-header">For sequence data:</li>
              <!-- <li class="divider"></li>   -->
              <li><a href="PopDiffSequenceData.html">Population differentiation</a></li>
          </ol>
        </li>
        <!-- Appendices dropdown -->
        <li class="dropdown">
          <a href="appendices" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Package Developers <span class="caret"></span></a>
          <ul class = "dropdown-menu" role = "menu">
           <li><a href = "PACKAGES.html">Population genetic packages in R</a></li>  
           <li><a href = "DATAFORMATS.html">Population genetic data types in R</a></li>
           <li><a href = "PACKAGE_TUTORIALS.html">Tutorials for developing R packages</a></li>
           <li class="divider"></li>  
           <li class="dropdown-header">Development Resources (external):</li>
           <li><a href = "http://r-pkgs.had.co.nz/">Hadley Wickham's guide for creating R packages</a></li>
           <li><a href = "https://github.com/jtleek/rpackages">Jeff Leek's R package development</a></li>
           <li><a href = "http://adv-r.had.co.nz/S4.html">S4 Classes in detail</a></li>
           <li><a href = "http://adv-r.had.co.nz/S3.html">S3 Classes in detail</a></li>
           <li><a href = "http://cran.r-project.org/doc/contrib/Genolini-S4tutorialV0-5en.pdf">A not so short introduction to S4</a></li>
          </ul>
        </li>
        <!-- Author button -->
        <li class="dropdown">
          <a href="about" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Contribute! <span class="caret"></span></a>
          <ul class = "dropdown-menu" role = "menu">
           <li><a href = "CONTRIBUTING.html">Contribution guidelines</a></li>  
           <li><a href = "CONTRIBUTING_WITH_GIT2R.html">Contributing with git2r</a></li> 
           <li class="divider"></li>  
           <li><a href="https://github.com/NESCent/popgenInfo">Source repository</a></li>  
		  </ul>
        </li>
        <li class = "button">
          <a href="LinksToResources.html">Useful Links</a>
        </li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<!-- <div class="container">
<div class="row">
<div class="span10 offset1"> -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Calculating Basic Population Genetic Statistics from SNP Data</h1>

</div>

<div id="TOC">
<ul>
<li><a href="#introduction">Introduction</a></li>
<li><a href="#load-necessary-resources-and-packages">Load necessary resources and packages</a></li>
<li><a href="#workflow-for-snp-data">Workflow for SNP data</a><ul>
<li><a href="#import-data">Import data</a></li>
<li><a href="#genetic-diversity-observed-and-expected-heterozygosity">Genetic diversity (observed and expected heterozygosity)</a></li>
<li><a href="#testing-for-hardy-weinberg-equilibrium">Testing for Hardy-Weinberg Equilibrium</a></li>
</ul></li>
<li><a href="#conclusion">Conclusion</a><ul>
<li><a href="#what-did-we-learn-today">What did we learn today?</a></li>
<li><a href="#what-is-next">What is next?</a></li>
</ul></li>
<li><a href="#references">References</a></li>
<li><a href="#contributors">Contributors</a></li>
<li><a href="#session-information">Session Information</a></li>
</ul>
</div>

<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>In this vignette, you will calculate basic population genetic statistics from SNP data using R packages. These statistics serve as exploratory analysis and require to work at the population level. We will calculate:</p>
<ol style="list-style-type: decimal">
<li>Genetic diversity,</li>
<li>Test Hardy Weinberg</li>
<li><span class="math inline">\(F_{is}\)</span> and global <span class="math inline">\(F_{st}\)</span>.</li>
</ol>
<p>The dataset used for these analyses are for lodgepole pine (<em>Pinus contorta</em>, Pinaceae), a plant species. More information on the dataset and the species is available from <a href="http://eckertdata.blogspot.fr/">A. Eckert’s website</a>. Here we use the dataset with no presupposed idea of interpreting the results in a biological way. To make the calculations faster, we work only with a subset of the full dataset.</p>
</div>
<div id="load-necessary-resources-and-packages" class="section level1">
<h1>Load necessary resources and packages</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;adegenet&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;hierfstat&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;pegas&quot;</span>)</code></pre></div>
</div>
<div id="workflow-for-snp-data" class="section level1">
<h1>Workflow for SNP data</h1>
<div id="import-data" class="section level2">
<h2>Import data</h2>
<p>The data are stored in a text file (genotype=AA..). We will import the dataset into R as a data frame, and then convert the SNP data file into a “<a href="http://www.inside-r.org/packages/cran/adegenet/docs/.valid.genind">genind</a>” object.</p>
<p>The dataset <a href="https://github.com/NESCent/popgenInfo/tree/master/data/Master_Pinus_data_genotype.txt">“Master_Pinus_data_genotype.txt” can be downloaded here</a>.</p>
<p>The text file is a matrix of (550 rows x 3086 columns). It contains 4 extra columns: first column is the label of the individuals, the three other are description of the region, all the other columns are for the genotypes as (AA or AT…).</p>
<p>When you import the data into R, the data file needs to be in your working directory, or adjust the path in the <code>read.table()</code> invocation below accordingly.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">Mydata &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;Master_Pinus_data_genotype.txt&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>)
<span class="kw">dim</span>(Mydata) <span class="co"># Matrix of dimension 550x3086</span></code></pre></div>
<pre><code>## [1]  550 3086</code></pre>
<p>To work with the data, we need to convert the R object returned by <code>read.table()</code> to a “genind” object. To achieve this, we create a matrix with only genotypes, and keep only a subset of the first 11 SNP loci (to make calculations faster). The result can then be converted to a “genind” object (for package <em>adegent</em>). The “genind” object can then easily be converted into a “hierfstat” (package <em>hierfstat</em>) object.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">locus &lt;-<span class="st"> </span>Mydata[, <span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">4</span>, <span class="dv">17</span><span class="op">:</span><span class="dv">3086</span>)]    
<span class="kw">colnames</span>(locus) &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">.&quot;</span>, <span class="st">&quot;_&quot;</span>, <span class="kw">colnames</span>(locus)) <span class="co"># locus names can&#39;t have &quot;.&quot;</span>
ind &lt;-<span class="st"> </span><span class="kw">as.character</span>(Mydata<span class="op">$</span>tree_id) <span class="co"># labels of the individuals</span>
population &lt;-<span class="st"> </span><span class="kw">as.character</span>(Mydata<span class="op">$</span>state) <span class="co"># labels of the populations</span>
Mydata1 &lt;-<span class="st"> </span><span class="kw">df2genind</span>(locus, <span class="dt">ploidy =</span> <span class="dv">2</span>, <span class="dt">ind.names =</span> ind, <span class="dt">pop =</span> population, <span class="dt">sep =</span> <span class="st">&quot;&quot;</span>)
Mydata1</code></pre></div>
<pre><code>## /// GENIND OBJECT /////////
## 
##  // 550 individuals; 12 loci; 24 alleles; size: 97.5 Kb
## 
##  // Basic content
##    @tab:  550 x 24 matrix of allele counts
##    @loc.n.all: number of alleles per locus (range: 2-2)
##    @loc.fac: locus factor for the 24 columns of @tab
##    @all.names: list of allele names for each locus
##    @ploidy: ploidy of each individual  (range: 2-2)
##    @type:  codom
##    @call: df2genind(X = locus, sep = &quot;&quot;, ind.names = ind, pop = population, 
##     ploidy = 2)
## 
##  // Optional content
##    @pop: population of each individual (group size range: 4-177)</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">nAll</span>(Mydata1) <span class="co"># Number of alleles per locus</span></code></pre></div>
<pre><code>## X0_10037_01_257 X0_10040_02_394 X0_10044_01_392  X0_10048_01_60 
##               2               2               2               2 
## X0_10051_02_166 X0_10054_01_402 X0_10067_03_111 X0_10079_02_168 
##               2               2               2               2 
## X0_10112_01_169 X0_10113_01_119 X0_10116_01_165  X0_10151_01_86 
##               2               2               2               2</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">Mydata2 &lt;-<span class="st"> </span><span class="kw">genind2hierfstat</span>(Mydata1) <span class="co"># Create hierfstat object</span></code></pre></div>
</div>
<div id="genetic-diversity-observed-and-expected-heterozygosity" class="section level2">
<h2>Genetic diversity (observed and expected heterozygosity)</h2>
<p>With <em>adegenet</em>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">div &lt;-<span class="st"> </span><span class="kw">summary</span>(Mydata1)
div</code></pre></div>
<pre><code>## 
## // Number of individuals: 550
## // Group sizes: 10 73 177 58 169 7 10 24 9 9 4
## // Number of alleles per locus: 2 2 2 2 2 2 2 2 2 2 2 2
## // Number of alleles per group: 22 24 24 23 24 20 19 23 21 19 18
## // Percentage of missing data: 1.62 %
## // Observed heterozygosity: 0.42 0.49 0.46 0.4 0.12 0.47 0.07 0.08 0.11 0.46 0.04 0.14
## // Expected heterozygosity: 0.41 0.5 0.5 0.45 0.13 0.5 0.07 0.08 0.11 0.44 0.04 0.16</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">names</span>(div)</code></pre></div>
<pre><code>## [1] &quot;n&quot;         &quot;n.by.pop&quot;  &quot;loc.n.all&quot; &quot;pop.n.all&quot; &quot;NA.perc&quot;   &quot;Hobs&quot;     
## [7] &quot;Hexp&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(div<span class="op">$</span>Hobs, <span class="dt">xlab=</span><span class="st">&quot;Loci number&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;Observed Heterozygosity&quot;</span>, 
     <span class="dt">main=</span><span class="st">&quot;Observed heterozygosity per locus&quot;</span>)</code></pre></div>
<p><img src="StartSNP_files/figure-html/Genetic_Diversity-1.png" width="960" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(div<span class="op">$</span>Hobs,div<span class="op">$</span>Hexp, <span class="dt">xlab=</span><span class="st">&quot;Hobs&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;Hexp&quot;</span>, 
     <span class="dt">main=</span><span class="st">&quot;Expected heterozygosity as a function of observed heterozygosity per locus&quot;</span>)</code></pre></div>
<p><img src="StartSNP_files/figure-html/Genetic_Diversity-2.png" width="960" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">bartlett.test</span>(<span class="kw">list</span>(div<span class="op">$</span>Hexp, div<span class="op">$</span>Hobs)) <span class="co"># a test : H0: Hexp = Hobs</span></code></pre></div>
<pre><code>## 
##  Bartlett test of homogeneity of variances
## 
## data:  list(div$Hexp, div$Hobs)
## Bartlett&#39;s K-squared = 0.011457, df = 1, p-value = 0.9148</code></pre>
<p>We observed that heterozygosity varies among loci. We observed no difference between expected and observed heterozygosity.</p>
<p>Basic statistics with <em>hierfstat</em>. Populations are states. The function <code>basic.stats()</code> provides the observed heterozygosity (<span class="math inline">\(H_o\)</span>), mean gene diversities within population (<span class="math inline">\(H_s\)</span>), <span class="math inline">\(F_{is}\)</span>, and <span class="math inline">\(F_{st}\)</span>. The function <code>boot.ppfis()</code> provides confidence interval for <span class="math inline">\(F_{is}\)</span>. The function <code>indpca()</code> does an PCA on the centered matrix of individuals’ allele frequencies.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">basicstat &lt;-<span class="st"> </span><span class="kw">basic.stats</span>(Mydata2, <span class="dt">diploid =</span> <span class="ot">TRUE</span>, <span class="dt">digits =</span> <span class="dv">2</span>) 
<span class="kw">names</span>(basicstat)   </code></pre></div>
<pre><code>## [1] &quot;n.ind.samp&quot; &quot;pop.freq&quot;   &quot;Ho&quot;         &quot;Hs&quot;         &quot;Fis&quot;       
## [6] &quot;perloc&quot;     &quot;overall&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">boot.ppfis</span>(Mydata2) </code></pre></div>
<pre><code>## $call
## boot.ppfis(dat = Mydata2)
## 
## $fis.ci
##         ll      hl
## 1  -0.0708  0.2875
## 2  -0.0675  0.0360
## 3   0.0117  0.0900
## 4  -0.0579  0.0718
## 5  -0.0408  0.0822
## 6  -0.4722  0.1852
## 7  -0.2609  0.1636
## 8  -0.2122 -0.0582
## 9  -0.1484  0.3040
## 10  0.0120  0.3153
## 11 -0.4348 -0.1633</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">x &lt;-<span class="st"> </span><span class="kw">indpca</span>(Mydata2) 
<span class="kw">plot</span>(x, <span class="dt">cex =</span> <span class="fl">0.7</span>)</code></pre></div>
<p><img src="StartSNP_files/figure-html/basic_statistics-1.png" width="960" /></p>
</div>
<div id="testing-for-hardy-weinberg-equilibrium" class="section level2">
<h2>Testing for Hardy-Weinberg Equilibrium</h2>
<p>We used the function <code>hw.test()</code> from the <em>pegas</em> package.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">hw.test</span>(Mydata1, <span class="dt">B =</span> <span class="dv">1000</span>)</code></pre></div>
<pre><code>##                      chi^2 df Pr(chi^2 &gt;) Pr.exact
## X0_10037_01_257 0.75465490  1 0.385006451    0.457
## X0_10040_02_394 0.16628227  1 0.683437231    0.723
## X0_10044_01_392 2.96053195  1 0.085319867    0.089
## X0_10048_01_60  6.39076800  1 0.011471539    0.018
## X0_10051_02_166 1.12035389  1 0.289842265    0.319
## X0_10054_01_402 1.54582494  1 0.213752853    0.251
## X0_10067_03_111 0.04868128  1 0.825374035    0.567
## X0_10079_02_168 0.01579313  1 0.899992594    0.603
## X0_10112_01_169 0.41126025  1 0.521330544    1.000
## X0_10113_01_119 0.74248768  1 0.388865151    0.442
## X0_10116_01_165 1.87098306  1 0.171362531    0.242
## X0_10151_01_86  8.94232481  1 0.002786379    0.005</code></pre>
<p>We get for each locus a test of significance of the null hypothesis: <span class="math inline">\(H_0\)</span> the locus is in HW equilibrium in the population. All but one locus are in HW equilibrium.</p>
</div>
</div>
<div id="conclusion" class="section level1">
<h1>Conclusion</h1>
<div id="what-did-we-learn-today" class="section level2">
<h2>What did we learn today?</h2>
<p>In this vignette, we learned how to explore the patterns of genetic diversity and how to estimate the level of genetic differentiation in one population. Also, you have an idea of potential violations of the dataset to the null Wright-Fischer model.</p>
</div>
<div id="what-is-next" class="section level2">
<h2>What is next?</h2>
<p>You may now want to move into looking into population differentiation in more detail (See <a href="DifferentiationSNP.html">Calculating genetic differentiation and clustering methods from SNP data</a>)</p>
</div>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<p>Eckert, A. J., A. D. Bower, S. C. González-Martínez, J. L. Wegrzyn, G. Coop and D. B. Neale. 2010. Back to nature: Ecological genomics of loblolly pine (Pinus taeda, Pinaceae). Molecular Ecology 19: 3789-3805. <a href="doi:%5B10.1111/j.1365-294X.2010.04698.x%5D(http://dx.doi.org/10.1111/j.1365-294X.2010.04698.x)" class="uri">doi:[10.1111/j.1365-294X.2010.04698.x](http://dx.doi.org/10.1111/j.1365-294X.2010.04698.x)</a></p>
</div>
<div id="contributors" class="section level1">
<h1>Contributors</h1>
<ul>
<li>Stéphanie Manel (Author)</li>
<li>Zhian Kamvar (edits)</li>
</ul>
</div>
<div id="session-information" class="section level1">
<h1>Session Information</h1>
<p>This shows us useful information for reproducibility. Of particular importance are the versions of R and the packages used to create this workflow. It is considered good practice to record this information with every analysis.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">width =</span> <span class="dv">100</span>)
devtools<span class="op">::</span><span class="kw">session_info</span>()</code></pre></div>
<pre><code>## Session info --------------------------------------------------------------------------------------</code></pre>
<pre><code>##  setting  value                       
##  version  R version 3.4.3 (2017-11-30)
##  system   x86_64, linux-gnu           
##  ui       X11                         
##  language (EN)                        
##  collate  en_US.UTF-8                 
##  tz       UTC                         
##  date     2018-04-03</code></pre>
<pre><code>## Packages ------------------------------------------------------------------------------------------</code></pre>
<pre><code>##  package    * version   date       source        
##  ade4       * 1.7-10    2017-12-15 CRAN (R 3.4.3)
##  adegenet   * 2.1.1     2018-02-02 CRAN (R 3.4.3)
##  ape        * 5.0       2017-10-30 CRAN (R 3.4.3)
##  assertthat   0.2.0     2017-04-11 CRAN (R 3.4.3)
##  backports    1.1.2     2017-12-13 CRAN (R 3.4.3)
##  base       * 3.4.3     2018-02-21 local         
##  bindr        0.1       2016-11-13 CRAN (R 3.4.3)
##  bindrcpp     0.2       2017-06-17 CRAN (R 3.4.3)
##  boot         1.3-20    2017-07-30 CRAN (R 3.4.3)
##  cluster      2.0.6     2017-03-16 CRAN (R 3.4.3)
##  coda         0.19-1    2016-12-08 CRAN (R 3.4.3)
##  colorspace   1.3-2     2016-12-14 CRAN (R 3.4.3)
##  compiler     3.4.3     2018-02-21 local         
##  datasets   * 3.4.3     2018-02-21 local         
##  deldir       0.1-14    2017-04-22 CRAN (R 3.4.3)
##  devtools     1.13.5    2018-02-18 CRAN (R 3.4.3)
##  digest       0.6.15    2018-01-28 CRAN (R 3.4.3)
##  dplyr        0.7.4     2017-09-28 CRAN (R 3.4.3)
##  evaluate     0.10.1    2017-06-24 CRAN (R 3.4.3)
##  expm         0.999-2   2017-03-29 CRAN (R 3.4.3)
##  gdata        2.18.0    2017-06-06 CRAN (R 3.4.3)
##  ggplot2      2.2.1     2016-12-30 CRAN (R 3.4.3)
##  glue         1.2.0     2017-10-29 CRAN (R 3.4.3)
##  gmodels      2.16.2    2015-07-22 CRAN (R 3.4.3)
##  graphics   * 3.4.3     2018-02-21 local         
##  grDevices  * 3.4.3     2018-02-21 local         
##  grid         3.4.3     2018-02-21 local         
##  gtable       0.2.0     2016-02-26 CRAN (R 3.4.3)
##  gtools       3.5.0     2015-05-29 CRAN (R 3.4.3)
##  hierfstat  * 0.04-22   2015-12-04 CRAN (R 3.4.3)
##  htmltools    0.3.6     2017-04-28 CRAN (R 3.4.3)
##  httpuv       1.3.5     2017-07-04 CRAN (R 3.4.3)
##  igraph       1.1.2     2017-07-21 CRAN (R 3.4.3)
##  knitr        1.20      2018-02-20 CRAN (R 3.4.3)
##  lattice      0.20-35   2017-03-25 CRAN (R 3.4.3)
##  lazyeval     0.2.1     2017-10-29 CRAN (R 3.4.3)
##  LearnBayes   2.15      2014-05-29 CRAN (R 3.4.3)
##  magrittr     1.5       2014-11-22 CRAN (R 3.4.3)
##  MASS         7.3-48    2017-12-25 CRAN (R 3.4.3)
##  Matrix       1.2-12    2017-11-16 CRAN (R 3.4.3)
##  memoise      1.1.0     2017-04-21 CRAN (R 3.4.3)
##  methods    * 3.4.3     2018-02-21 local         
##  mgcv         1.8-23    2018-01-15 CRAN (R 3.4.3)
##  mime         0.5       2016-07-07 CRAN (R 3.4.3)
##  munsell      0.4.3     2016-02-13 CRAN (R 3.4.3)
##  nlme         3.1-131.1 2018-02-16 CRAN (R 3.4.3)
##  parallel     3.4.3     2018-02-21 local         
##  pegas      * 0.10      2017-05-03 CRAN (R 3.4.3)
##  permute      0.9-4     2016-09-09 CRAN (R 3.4.3)
##  pillar       1.1.0     2018-01-14 CRAN (R 3.4.3)
##  pkgconfig    2.0.1     2017-03-21 CRAN (R 3.4.3)
##  plyr         1.8.4     2016-06-08 CRAN (R 3.4.3)
##  R6           2.2.2     2017-06-17 CRAN (R 3.4.3)
##  Rcpp         0.12.15   2018-01-20 CRAN (R 3.4.3)
##  reshape2     1.4.3     2017-12-11 CRAN (R 3.4.3)
##  rlang        0.2.0     2018-02-20 CRAN (R 3.4.3)
##  rmarkdown    1.8       2017-11-17 CRAN (R 3.4.3)
##  rprojroot    1.3-2     2018-01-03 CRAN (R 3.4.3)
##  scales       0.5.0     2017-08-24 CRAN (R 3.4.3)
##  seqinr       3.4-5     2017-08-01 CRAN (R 3.4.3)
##  shiny        1.0.5     2017-08-23 CRAN (R 3.4.3)
##  sp           1.2-7     2018-01-19 CRAN (R 3.4.3)
##  spData       0.2.7.4   2018-02-11 CRAN (R 3.4.3)
##  spdep        0.7-4     2017-11-22 CRAN (R 3.4.3)
##  splines      3.4.3     2018-02-21 local         
##  stats      * 3.4.3     2018-02-21 local         
##  stringi      1.1.6     2017-11-17 CRAN (R 3.4.3)
##  stringr      1.3.0     2018-02-19 CRAN (R 3.4.3)
##  tibble       1.4.2     2018-01-22 CRAN (R 3.4.3)
##  tools        3.4.3     2018-02-21 local         
##  utils      * 3.4.3     2018-02-21 local         
##  vegan        2.4-6     2018-01-24 CRAN (R 3.4.3)
##  withr        2.1.1     2017-12-19 CRAN (R 3.4.3)
##  xtable       1.8-2     2016-02-05 CRAN (R 3.4.3)
##  yaml         2.1.16    2017-12-12 CRAN (R 3.4.3)</code></pre>
</div>

<!-- </div>
</div>
</div>
 -->



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
