<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Calculating Basic Population Genetic Statistics from Microsatellite Data</title>

<script src="libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.5/css/spacelab.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<!---

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
  padding-right: 50px;
  padding-left: -50px;
}
@media (max-width: 979px) {
  body {
    padding-top: 0;
  }
}

 offset scroll position for anchor links (for fixed navbar)  
@media (min-width: 980px) {
  .section h2 {
    padding-top: 52px;
    margin-top: -52px;
  }
  .section h3 {
    padding-top: 52px;
    margin-top: -52px;
  }
}


/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

</style>
-->



<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}
/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}
/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}
</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #f8f8f8; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
pre, code { background-color: #f8f8f8; }
code > span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code > span.dt { color: #204a87; } /* DataType */
code > span.dv { color: #0000cf; } /* DecVal */
code > span.bn { color: #0000cf; } /* BaseN */
code > span.fl { color: #0000cf; } /* Float */
code > span.ch { color: #4e9a06; } /* Char */
code > span.st { color: #4e9a06; } /* String */
code > span.co { color: #8f5902; font-style: italic; } /* Comment */
code > span.ot { color: #8f5902; } /* Other */
code > span.al { color: #ef2929; } /* Alert */
code > span.fu { color: #000000; } /* Function */
code > span.er { color: #a40000; font-weight: bold; } /* Error */
code > span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #000000; } /* Constant */
code > span.sc { color: #000000; } /* SpecialChar */
code > span.vs { color: #4e9a06; } /* VerbatimString */
code > span.ss { color: #4e9a06; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #000000; } /* Variable */
code > span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code > span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code > span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code > span.ex { } /* Extension */
code > span.at { color: #c4a000; } /* Attribute */
code > span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code > span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<div class="container-fluid main-container">

<!-- tabsets -->
<script src="libs/navigation-1.0/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <!-- HOME -->
      <a class="navbar-brand" href="index.html"><b>Population Genetics in R</b></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <!-- Table of contents -->
        <li class="dropdown">
          <a href="chapters" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Users <span class="caret"></span></a>
          <ol class="dropdown-menu">
              <li><a href="GETSTARTED.html">Getting Started</a></li>
              <li class="divider"></li>
              <li class="dropdown-header">For microsatellite data:</li>
              <li><a href="startMicrosatellite.html">Basic statistics</a></li>
              <li><a href="2015-12-15-microsatellite-differentiation.html">Population differentiation</a></li>
              <li class="divider"></li>
              <li class="dropdown-header">For SNP data:</li>
              <!-- <li class="divider"></li> -->
              <li><a href="StartSNP.html">Basic statistics</a></li>
              <li><a href="DifferentiationSNP.html">Population differentiation</a></li>
              <li><a href="2015-05-18-Dist-SNP.html">Genetic distance (individual-based)</a></li>
              <li><a href="2016-01-26-SNP-selection.html">Signals of selection</a></li>
              <!-- <li><a href="SNP.html">SNPs</a></li> -->
              <li class="divider"></li>
              <li class="dropdown-header">For sequence data:</li>
              <!-- <li class="divider"></li>   -->
              <li><a href="PopDiffSequenceData.html">Population differentiation</a></li>
          </ol>
        </li>
        <!-- Appendices dropdown -->
        <li class="dropdown">
          <a href="appendices" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Package Developers <span class="caret"></span></a>
          <ul class = "dropdown-menu" role = "menu">
           <li><a href = "PACKAGES.html">Population genetic packages in R</a></li>  
           <li><a href = "DATAFORMATS.html">Population genetic data types in R</a></li>
           <li><a href = "PACKAGE_TUTORIALS.html">Tutorials for developing R packages</a></li>
           <li class="divider"></li>  
           <li class="dropdown-header">Development Resources (external):</li>
           <li><a href = "http://r-pkgs.had.co.nz/">Hadley Wickham's guide for creating R packages</a></li>
           <li><a href = "https://github.com/jtleek/rpackages">Jeff Leek's R package development</a></li>
           <li><a href = "http://adv-r.had.co.nz/S4.html">S4 Classes in detail</a></li>
           <li><a href = "http://adv-r.had.co.nz/S3.html">S3 Classes in detail</a></li>
           <li><a href = "http://cran.r-project.org/doc/contrib/Genolini-S4tutorialV0-5en.pdf">A not so short introduction to S4</a></li>
          </ul>
        </li>
        <!-- Author button -->
        <li class="dropdown">
          <a href="about" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Contribute! <span class="caret"></span></a>
          <ul class = "dropdown-menu" role = "menu">
           <li><a href = "CONTRIBUTING.html">Contribution guidelines</a></li>  
           <li><a href = "CONTRIBUTING_WITH_GIT2R.html">Contributing with git2r</a></li> 
           <li class="divider"></li>  
           <li><a href="https://github.com/NESCent/popgenInfo">Source repository</a></li>  
		  </ul>
        </li>
        <li class = "button">
          <a href="LinksToResources.html">Useful Links</a>
        </li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<!-- <div class="container">
<div class="row">
<div class="span10 offset1"> -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Calculating Basic Population Genetic Statistics from Microsatellite Data</h1>

</div>

<div id="TOC">
<ul>
<li><a href="#introduction">Introduction</a></li>
<li><a href="#assumptions">Assumptions</a></li>
<li><a href="#data">Data</a></li>
<li><a href="#resourcespackages-required">Resources/Packages required</a></li>
<li><a href="#analysis">Analysis</a><ul>
<li><a href="#section-1-load-the-data">Section 1: Load the data</a></li>
<li><a href="#section-2-genetic-diversity-observed-and-expected-heterozygosity">Section 2: Genetic diversity (observed and expected heterozygosity)</a></li>
<li><a href="#section-3-basic-statistics-with-hierfstat">Section 3: Basic statistics with <em>hierfstat</em></a></li>
<li><a href="#section-4-testing-for-hardy-weinberg-equilibrium">Section 4: Testing for Hardy-Weinberg Equilibrium</a></li>
</ul></li>
<li><a href="#conclusions">Conclusions</a><ul>
<li><a href="#what-did-we-learn-today">What did we learn today?</a></li>
<li><a href="#what-is-next">What is next?</a></li>
</ul></li>
<li><a href="#contributors">Contributors</a></li>
<li><a href="#session-information">Session Information</a></li>
</ul>
</div>

<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>In this vignette, you will calculate basic population genetic statistics from microsatellite data using R packages.</p>
<ol style="list-style-type: decimal">
<li>Genetic diversity,</li>
<li>Test Hardy Weinberg</li>
<li><span class="math inline">\(F_{is}\)</span> and global <span class="math inline">\(F_{st}\)</span></li>
</ol>
</div>
<div id="assumptions" class="section level1">
<h1>Assumptions</h1>
<ul>
<li><span class="math inline">\(H_0\)</span>: we assume that the microsatellites are in Hardy-Weinberg equilibrium</li>
<li><span class="math inline">\(H_1\)</span>: they are not.</li>
</ul>
</div>
<div id="data" class="section level1">
<h1>Data</h1>
<p>The data used for these analyses are contained in an R dataset: <code>nancycats</code>, a “genind” object, i.e. an object of the package <em>adegenet</em>. It contains microsatellite genotypes of 237 cats from 17 colonies of Nancy (France). Each individuals of the 17 colonies are genotyped for 9 microsatellite loci.</p>
</div>
<div id="resourcespackages-required" class="section level1">
<h1>Resources/Packages required</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;adegenet&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;pegas&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;hierfstat&quot;</span>)</code></pre></div>
</div>
<div id="analysis" class="section level1">
<h1>Analysis</h1>
<div id="section-1-load-the-data" class="section level2">
<h2>Section 1: Load the data</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(<span class="st">&quot;nancycats&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;adegenet&quot;</span>)  
<span class="kw">summary</span>(nancycats)</code></pre></div>
<pre><code>## 
## // Number of individuals: 237
## // Group sizes: 10 22 12 23 15 11 14 10 9 11 20 14 13 17 11 12 13
## // Number of alleles per locus: 16 11 10 9 12 8 12 12 18
## // Number of alleles per group: 36 53 50 67 48 56 42 54 43 46 70 52 44 61 42 40 35
## // Percentage of missing data: 2.34 %
## // Observed heterozygosity: 0.67 0.67 0.68 0.71 0.63 0.57 0.65 0.62 0.45
## // Expected heterozygosity: 0.87 0.79 0.8 0.76 0.87 0.69 0.82 0.76 0.61</code></pre>
</div>
<div id="section-2-genetic-diversity-observed-and-expected-heterozygosity" class="section level2">
<h2>Section 2: Genetic diversity (observed and expected heterozygosity)</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">nancycats  </code></pre></div>
<pre><code>## /// GENIND OBJECT /////////
## 
##  // 237 individuals; 9 loci; 108 alleles; size: 145.3 Kb
## 
##  // Basic content
##    @tab:  237 x 108 matrix of allele counts
##    @loc.n.all: number of alleles per locus (range: 8-18)
##    @loc.fac: locus factor for the 108 columns of @tab
##    @all.names: list of allele names for each locus
##    @ploidy: ploidy of each individual  (range: 2-2)
##    @type:  codom
##    @call: genind(tab = truenames(nancycats)$tab, pop = truenames(nancycats)$pop)
## 
##  // Optional content
##    @pop: population of each individual (group size range: 9-23)
##    @other: a list containing: xy</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">div &lt;-<span class="st"> </span><span class="kw">summary</span>(nancycats)
div</code></pre></div>
<pre><code>## 
## // Number of individuals: 237
## // Group sizes: 10 22 12 23 15 11 14 10 9 11 20 14 13 17 11 12 13
## // Number of alleles per locus: 16 11 10 9 12 8 12 12 18
## // Number of alleles per group: 36 53 50 67 48 56 42 54 43 46 70 52 44 61 42 40 35
## // Percentage of missing data: 2.34 %
## // Observed heterozygosity: 0.67 0.67 0.68 0.71 0.63 0.57 0.65 0.62 0.45
## // Expected heterozygosity: 0.87 0.79 0.8 0.76 0.87 0.69 0.82 0.76 0.61</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(div$Hobs, <span class="dt">xlab=</span><span class="st">&quot;Loci number&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;Observed Heterozygosity&quot;</span>, 
     <span class="dt">main=</span><span class="st">&quot;Observed heterozygosity per locus&quot;</span>)</code></pre></div>
<p><img src="startMicrosatellite_files/figure-html/Genetic_diversity-1.png" width="960" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(div$Hobs, div$Hexp, <span class="dt">xlab=</span><span class="st">&quot;Observed Heterozygosity&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;Expected Heterozygosity&quot;</span>, 
     <span class="dt">main=</span><span class="st">&quot;Expected heterozygosity as a function of observed heterozygosity per locus&quot;</span>)</code></pre></div>
<p><img src="startMicrosatellite_files/figure-html/Genetic_diversity-2.png" width="960" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">bartlett.test</span>(<span class="kw">list</span>(div$Hexp, div$Hobs)) <span class="co"># difference of means </span></code></pre></div>
<pre><code>## 
##  Bartlett test of homogeneity of variances
## 
## data:  list(div$Hexp, div$Hobs)
## Bartlett&#39;s K-squared = 0.046962, df = 1, p-value = 0.8284</code></pre>
<p>We get various information from the command summary. The one that interest us is the observed and expected heterozygosity per locus. We observed that heterozygosity varies among loci.</p>
<p>The results from the Bartlett test indicates that we have no difference between the mean observed and expected heterozygosity</p>
</div>
<div id="section-3-basic-statistics-with-hierfstat" class="section level2">
<h2>Section 3: Basic statistics with <em>hierfstat</em></h2>
<p>The function <code>basic.stats()</code> provides the observed heterozygosity (<span class="math inline">\(H_o\)</span>), mean gene diversities within population (<span class="math inline">\(H_s\)</span>), <span class="math inline">\(F_{is}\)</span>, and <span class="math inline">\(F_{st}\)</span>. The function <code>boot.ppfis()</code> provides confidence interval for <span class="math inline">\(F_{is}\)</span>. The function <code>indpca()</code> does a PCA on the centered matrix of individuals’ allele frequencies.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">nancycats.hfstat &lt;-<span class="st"> </span><span class="kw">genind2hierfstat</span>(nancycats)
basicstat &lt;-<span class="st"> </span><span class="kw">basic.stats</span>(nancycats, <span class="dt">diploid =</span> <span class="ot">TRUE</span>, <span class="dt">digits =</span> <span class="dv">2</span>) 
<span class="kw">names</span>(basicstat)</code></pre></div>
<pre><code>## [1] &quot;n.ind.samp&quot; &quot;pop.freq&quot;   &quot;Ho&quot;         &quot;Hs&quot;         &quot;Fis&quot;       
## [6] &quot;perloc&quot;     &quot;overall&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">boot.ppfis</span>(nancycats.hfstat) </code></pre></div>
<pre><code>## $call
## boot.ppfis(dat = nancycats.hfstat)
## 
## $fis.ci
##         ll     hl
## 1  -0.0091 0.3225
## 2   0.1125 0.2877
## 3   0.0717 0.2843
## 4   0.0836 0.2485
## 5   0.0432 0.2340
## 6   0.0684 0.3041
## 7   0.0777 0.2984
## 8   0.0578 0.3579
## 9  -0.2368 0.0739
## 10 -0.0621 0.2425
## 11  0.0855 0.2418
## 12  0.0701 0.2647
## 13 -0.1090 0.1261
## 14  0.0021 0.2395
## 15 -0.1753 0.0812
## 16 -0.0604 0.1725
## 17 -0.1403 0.1928</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">x &lt;-<span class="st"> </span><span class="kw">indpca</span>(nancycats.hfstat) 
<span class="kw">plot</span>(x, <span class="dt">cex =</span> <span class="fl">0.7</span>)</code></pre></div>
<p><img src="startMicrosatellite_files/figure-html/basic_statistics-1.png" width="960" /></p>
</div>
<div id="section-4-testing-for-hardy-weinberg-equilibrium" class="section level2">
<h2>Section 4: Testing for Hardy-Weinberg Equilibrium</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">hw.test</span>(nancycats, <span class="dt">B =</span> <span class="dv">1000</span>)</code></pre></div>
<pre><code>##           chi^2  df  Pr(chi^2 &gt;) Pr.exact
## fca8  395.80006 120 0.000000e+00        0
## fca23 239.34221  55 0.000000e+00        0
## fca43 434.33397  45 0.000000e+00        0
## fca45  66.11849  36 1.622163e-03        0
## fca77 270.52066  66 0.000000e+00        0
## fca78 402.80002  28 0.000000e+00        0
## fca90 217.19836  66 0.000000e+00        0
## fca96 193.36764  66 1.965095e-14        0
## fca37 291.00731 153 1.209777e-10        0</code></pre>
<p>We get for each locus a test of significance of the null hypothesis: <span class="math inline">\(H_0\)</span> the locus is in HW equilibrium in the population/ <span class="math inline">\(H_1\)</span>. The locus is not in HW equilibrium.</p>
<p>We can conclude from the p-values of each test that any locus is in HW equilibrium.</p>
</div>
</div>
<div id="conclusions" class="section level1">
<h1>Conclusions</h1>
<div id="what-did-we-learn-today" class="section level2">
<h2>What did we learn today?</h2>
<p>In this vignette, we learned how to explore the patterns of genetic diversity in one population. Also, you have an idea of potential violations of the dataset to the null Wright-Fischer model.</p>
</div>
<div id="what-is-next" class="section level2">
<h2>What is next?</h2>
<p>You may now want to move on to looking into <a href="2015-12-15-microsatellite-differentiation.html">population differentiation</a>.</p>
</div>
</div>
<div id="contributors" class="section level1">
<h1>Contributors</h1>
<ul>
<li>Stéphanie Manel (Author)</li>
<li>Zhian Kamvar (edits)</li>
</ul>
</div>
<div id="session-information" class="section level1">
<h1>Session Information</h1>
<p>This shows us useful information for reproducibility. Of particular importance are the versions of R and the packages used to create this workflow. It is considered good practice to record this information with every analysis.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">width =</span> <span class="dv">100</span>)
devtools::<span class="kw">session_info</span>()</code></pre></div>
<pre><code>## Session info ---------------------------------------------------------------------------------------</code></pre>
<pre><code>##  setting  value                       
##  version  R version 3.3.0 (2016-05-03)
##  system   x86_64, linux-gnu           
##  ui       X11                         
##  language (EN)                        
##  collate  en_US.UTF-8                 
##  tz       &lt;NA&gt;                        
##  date     2016-05-24</code></pre>
<pre><code>## Packages -------------------------------------------------------------------------------------------</code></pre>
<pre><code>##  package    * version date       source        
##  ade4       * 1.7-4   2016-03-01 CRAN (R 3.3.0)
##  adegenet   * 2.0.1   2016-02-15 CRAN (R 3.3.0)
##  ape        * 3.4     2015-11-29 CRAN (R 3.3.0)
##  assertthat   0.1     2013-12-06 CRAN (R 3.3.0)
##  boot         1.3-18  2016-02-23 CRAN (R 3.3.0)
##  cluster      2.0.4   2016-04-18 CRAN (R 3.2.5)
##  coda         0.18-1  2015-10-16 CRAN (R 3.3.0)
##  colorspace   1.2-6   2015-03-11 CRAN (R 3.3.0)
##  DBI          0.4-1   2016-05-08 CRAN (R 3.3.0)
##  deldir       0.1-12  2016-03-06 CRAN (R 3.3.0)
##  devtools     1.11.1  2016-04-21 CRAN (R 3.3.0)
##  digest       0.6.9   2016-01-08 CRAN (R 3.3.0)
##  dplyr        0.4.3   2015-09-01 CRAN (R 3.3.0)
##  evaluate     0.9     2016-04-29 CRAN (R 3.3.0)
##  formatR      1.4     2016-05-09 CRAN (R 3.3.0)
##  gdata        2.17.0  2015-07-04 CRAN (R 3.3.0)
##  ggplot2      2.1.0   2016-03-01 CRAN (R 3.3.0)
##  gmodels      2.16.2  2015-07-22 CRAN (R 3.3.0)
##  gtable       0.2.0   2016-02-26 CRAN (R 3.3.0)
##  gtools       3.5.0   2015-05-29 CRAN (R 3.3.0)
##  hierfstat  * 0.04-22 2015-12-04 CRAN (R 3.3.0)
##  htmltools    0.3.5   2016-03-21 CRAN (R 3.3.0)
##  httpuv       1.3.3   2015-08-04 CRAN (R 3.3.0)
##  igraph       1.0.1   2015-06-26 CRAN (R 3.3.0)
##  knitr        1.13    2016-05-09 CRAN (R 3.3.0)
##  lattice      0.20-33 2015-07-14 CRAN (R 3.3.0)
##  LearnBayes   2.15    2014-05-29 CRAN (R 3.3.0)
##  magrittr     1.5     2014-11-22 CRAN (R 3.3.0)
##  MASS         7.3-45  2016-04-21 CRAN (R 3.3.0)
##  Matrix       1.2-6   2016-05-02 CRAN (R 3.3.0)
##  memoise      1.0.0   2016-01-29 CRAN (R 3.3.0)
##  mgcv         1.8-12  2016-03-03 CRAN (R 3.3.0)
##  mime         0.4     2015-09-03 CRAN (R 3.3.0)
##  munsell      0.4.3   2016-02-13 CRAN (R 3.3.0)
##  nlme         3.1-128 2016-05-10 CRAN (R 3.3.0)
##  pegas      * 0.9     2016-04-16 CRAN (R 3.3.0)
##  permute      0.9-0   2016-01-24 CRAN (R 3.3.0)
##  plyr         1.8.3   2015-06-12 CRAN (R 3.3.0)
##  R6           2.1.2   2016-01-26 CRAN (R 3.3.0)
##  Rcpp         0.12.5  2016-05-14 CRAN (R 3.3.0)
##  reshape2     1.4.1   2014-12-06 CRAN (R 3.3.0)
##  rmarkdown    0.9.6   2016-05-01 CRAN (R 3.3.0)
##  scales       0.4.0   2016-02-26 CRAN (R 3.3.0)
##  seqinr       3.1-3   2014-12-17 CRAN (R 3.3.0)
##  shiny        0.13.2  2016-03-28 CRAN (R 3.3.0)
##  sp           1.2-3   2016-04-14 CRAN (R 3.3.0)
##  spdep        0.6-4   2016-04-12 CRAN (R 3.3.0)
##  stringi      1.0-1   2015-10-22 CRAN (R 3.3.0)
##  stringr      1.0.0   2015-04-30 CRAN (R 3.3.0)
##  vegan        2.3-5   2016-04-09 CRAN (R 3.3.0)
##  withr        1.0.1   2016-02-04 CRAN (R 3.3.0)
##  xtable       1.8-2   2016-02-05 CRAN (R 3.3.0)
##  yaml         2.1.13  2014-06-12 CRAN (R 3.3.0)</code></pre>
</div>

<!-- </div>
</div>
</div>
 -->



</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
