<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />



<title>Template Vignette</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/spacelab.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>
<!---

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
  padding-right: 50px;
  padding-left: -50px;
}
@media (max-width: 979px) {
  body {
    padding-top: 0;
  }
}

 offset scroll position for anchor links (for fixed navbar)  
@media (min-width: 980px) {
  .section h2 {
    padding-top: 52px;
    margin-top: -52px;
  }
  .section h3 {
    padding-top: 52px;
    margin-top: -52px;
  }
}


/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

</style>
-->



<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}
/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}
/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}
</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #f8f8f8; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
pre, code { background-color: #f8f8f8; }
code > span.kw { color: #204a87; font-weight: bold; }
code > span.dt { color: #204a87; }
code > span.dv { color: #0000cf; }
code > span.bn { color: #0000cf; }
code > span.fl { color: #0000cf; }
code > span.ch { color: #4e9a06; }
code > span.st { color: #4e9a06; }
code > span.co { color: #8f5902; font-style: italic; }
code > span.ot { color: #8f5902; }
code > span.al { color: #ef2929; }
code > span.fu { color: #000000; }
code > span.er { font-weight: bold; }
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <!-- HOME -->
      <a class="navbar-brand" href="index.html"><b>Population Genetics in R</b></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <!-- Table of contents -->
        <li class="dropdown">
          <a href="chapters" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Users <span class="caret"></span></a>
          <ol class="dropdown-menu">
              <li><a href="GETSTARTED.html">Getting Started</a></li>
              <li class="divider"></li>
              <li class="dropdown-header">For SNP data:</li>
              <!-- <li class="divider"></li> -->
              <li><a href="StartSNP.html">Basic statistics</a></li>
              <li><a href="DifferentiationSNP.html">Population differentiation</a></li>
              <li><a href="VignetteIndvidualGeneticDistancesSNP.html">Genetic Distance (individual-based)</a></li>
              <!-- <li><a href="SNP.html">SNPs</a></li> -->
              <li class="divider"></li>
              <li class="dropdown-header">For sequence data:</li>
              <!-- <li class="divider"></li>   -->
              <li><a href="SummarySequenceData.html">Summary statistics</a></li>
              <li><a href="PopDiffSequenceData.html">Population differentiation</a></li>
          </ol>
        </li>
        <!-- Appendices dropdown -->
        <li class="dropdown">
          <a href="appendices" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Package Developers <span class="caret"></span></a>
          <ul class = "dropdown-menu", role = "menu">
           <li><a href = "DEVELOPER_INTRO.html">Introduction</a></li>
           <li><a href = "PACKAGES.html">Population genetic packages in R</a></li>  
           <li><a href = "DATAFORMATS.html">Population genetic data types in R</a></li>
           <li><a href = "PACKAGE_TUTORIALS.html">Developing R packages</a></li>
           <li class="divider"></li>  
           <li class="dropdown-header">Development Resources (external):</li>
           <li><a href = "http://r-pkgs.had.co.nz/">Hadley Wickham's Guide For Creating R Packages</a></li>
           <li><a href = "https://github.com/jtleek/rpackages">Jeff Leek's R Package Development</a></li>
           <li><a href = "http://adv-r.had.co.nz/S4.html">S4 Classes in detail</a></li>
           <li><a href = "http://adv-r.had.co.nz/S3.html">S3 Classes in detail</a></li>
           <li><a href = "http://cran.r-project.org/doc/contrib/Genolini-S4tutorialV0-5en.pdf">A not so short introduction to S4</a></li>
          </ul>
        </li>
        <!-- Author button -->
        <li class="dropdown">
          <a href="about" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Contribute! <span class="caret"></span></a>
          <ul class = "dropdown-menu", role = "menu">
           <li><a href = "CONTRIBUTING.html">How to contribute</a></li>  
           <li><a href="https://github.com/NESCent/popgenInfo">Source repository</a></li>  
		  </ul>
        </li>
        <li class = "button">
          <a href="LinksToResources.html">Useful Links</a>
        </li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<!-- <div class="container">
<div class="row">
<div class="span10 offset1"> -->

<div id="header">
<h1 class="title">Template Vignette</h1>
</div>

<div id="TOC">
<ul>
<li><a href="#introduction">Introduction</a></li>
<li><a href="#assumptions">Assumptions</a></li>
<li><a href="#data">Data</a></li>
<li><a href="#resourcespackages-required">Resources/Packages required</a></li>
<li><a href="#analysis-divided-into-sections">Analysis divided into sections</a><ul>
<li><a href="#section-1-load-the-data">Section 1: Load the data</a></li>
<li><a href="#section-2-exploratory-data-analysischecking-assumptions">Section 2: Exploratory Data analysis/Checking assumptions</a></li>
<li><a href="#section-3-summary-statistics">Section 3: Summary statistics</a></li>
</ul></li>
<li><a href="#conclusions">Conclusions</a><ul>
<li><a href="#whats-next">What’s next</a></li>
</ul></li>
<li><a href="#contributors">Contributors</a></li>
<li><a href="#session-information">Session Information</a></li>
</ul>
</div>

<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>Explain the purpose of the vignette.<br />Give a brief explanation of the questions that will be answered and some background information. These vignettes are written in <a href="http://rmarkdown.rstudio.com">Rmarkdown</a>. You can find <a href="http://rmarkdown.rstudio.com/authoring_basics.html">details about markdown here</a> and <a href="http://rmarkdown.rstudio.com/authoring_rcodechunks.html">details about code chunks here</a>.</p>
</div>
<div id="assumptions" class="section level1">
<h1>Assumptions</h1>
<p>All biological analyses have assumptions.</p>
<ul>
<li>Assumption 1</li>
<li>Assumption 2</li>
</ul>
</div>
<div id="data" class="section level1">
<h1>Data</h1>
<p>Links to information about the data if needed e.g. package vignettes. You can use an <a href="https://github.com/cran/adegenet/blob/master/inst/files/nancycats.gtx">external link</a> or you can place the data <a href="https://github.com/NESCent/popgenInfo/tree/master/data/nancycats.gtx">in the <code>data/</code> folder</a> (keep this &lt; 200kb).</p>
</div>
<div id="resourcespackages-required" class="section level1">
<h1>Resources/Packages required</h1>
<p>Packages required. Say where these can be found, can link to the list of packages on our repository page here also.</p>
<p>Loading the required packages:</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(hierfstat)
<span class="kw">library</span>(adegenet)</code></pre>
</div>
<div id="analysis-divided-into-sections" class="section level1">
<h1>Analysis divided into sections</h1>
<p>Example Sections</p>
<div id="section-1-load-the-data" class="section level2">
<h2>Section 1: Load the data</h2>
<pre class="sourceCode r"><code class="sourceCode r">nancycats &lt;-<span class="st"> </span><span class="kw">import2genind</span>(<span class="st">&quot;nancycats.gtx&quot;</span>)
<span class="kw">summary</span>(nancycats)</code></pre>
<pre><code>## 
##  Converting data from GENETIX to a genind object... 
## 
## ...done.</code></pre>
<pre><code>## 
##  # Total number of genotypes:  237 
## 
##  # Population sample sizes:  
##  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 
## 10 22 12 23 15 11 14 10  9 11 20 14 13 17 11 12 13 
## 
##  # Number of alleles per locus:  
## L1 L2 L3 L4 L5 L6 L7 L8 L9 
## 16 11 10  9 12  8 12 12 18 
## 
##  # Number of alleles per population:  
## 01 02 03 04 05 06 07 08 09 10 11 12 13 14 15 16 17 
## 36 53 50 67 48 56 42 54 43 46 70 52 44 61 42 40 35 
## 
##  # Percentage of missing data:  
## [1] 2.344116
## 
##  # Observed heterozygosity:  
##        L1        L2        L3        L4        L5        L6        L7 
## 0.6682028 0.6666667 0.6793249 0.7083333 0.6329114 0.5654008 0.6497890 
##        L8        L9 
## 0.6184211 0.4514768 
## 
##  # Expected heterozygosity:  
##        L1        L2        L3        L4        L5        L6        L7 
## 0.8657224 0.7928751 0.7953319 0.7603095 0.8702576 0.6884669 0.8157881 
##        L8        L9 
## 0.7603493 0.6062686</code></pre>
</div>
<div id="section-2-exploratory-data-analysischecking-assumptions" class="section level2">
<h2>Section 2: Exploratory Data analysis/Checking assumptions</h2>
</div>
<div id="section-3-summary-statistics" class="section level2">
<h2>Section 3: Summary statistics</h2>
</div>
</div>
<div id="conclusions" class="section level1">
<h1>Conclusions</h1>
<div id="whats-next" class="section level2">
<h2>What’s next</h2>
<p>Information on further analysis that could be done, other workflows <a href="DifferentiationSNP.html">can be linked as well</a> (note the HTML as opposed to Rmd link).</p>
</div>
</div>
<div id="contributors" class="section level1">
<h1>Contributors</h1>
<ul>
<li>Author 1</li>
<li>Author 2</li>
<li>Contributor 1 (role)</li>
</ul>
</div>
<div id="session-information" class="section level1">
<h1>Session Information</h1>
<p>Information about the R session used for this tutorial.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">options</span>(<span class="dt">width =</span> <span class="dv">100</span>)
devtools::<span class="kw">session_info</span>()</code></pre>
<pre><code>## Session info ---------------------------------------------------------------------------------------</code></pre>
<pre><code>##  setting  value                       
##  version  R version 3.1.3 (2015-03-09)
##  system   x86_64, linux-gnu           
##  ui       X11                         
##  language (EN)                        
##  collate  en_US.UTF-8                 
##  tz       &lt;NA&gt;</code></pre>
<pre><code>## Packages -------------------------------------------------------------------------------------------</code></pre>
<pre><code>##  package    * version date       source        
##  ade4         1.6-2   2013-11-20 CRAN (R 3.1.3)
##  adegenet     1.4-2   2014-05-13 CRAN (R 3.1.3)
##  ape        * 3.2     2014-12-05 CRAN (R 3.1.3)
##  colorspace * 1.2-6   2015-03-11 CRAN (R 3.1.3)
##  devtools   * 1.7.0   2015-01-17 CRAN (R 3.1.3)
##  digest     * 0.6.8   2014-12-31 CRAN (R 3.1.3)
##  evaluate   * 0.5.5   2014-04-29 CRAN (R 3.1.3)
##  formatR    * 1.1     2015-03-29 CRAN (R 3.1.3)
##  ggplot2    * 1.0.1   2015-03-17 CRAN (R 3.1.3)
##  gtable     * 0.1.2   2012-12-05 CRAN (R 3.1.3)
##  gtools       3.4.1   2014-05-28 CRAN (R 3.1.3)
##  hierfstat    0.04-14 2014-09-29 CRAN (R 3.1.3)
##  htmltools  * 0.2.6   2014-09-08 CRAN (R 3.1.3)
##  httpuv     * 1.3.2   2014-10-23 CRAN (R 3.1.3)
##  igraph     * 0.7.1   2014-04-22 CRAN (R 3.1.3)
##  knitr      * 1.9     2015-01-20 CRAN (R 3.1.3)
##  lattice    * 0.20-31 2015-03-30 CRAN (R 3.1.3)
##  MASS       * 7.3-40  2015-03-21 CRAN (R 3.1.3)
##  mime       * 0.3     2015-03-29 CRAN (R 3.1.3)
##  munsell    * 0.4.2   2013-07-11 CRAN (R 3.1.3)
##  nlme       * 3.1-120 2015-02-20 CRAN (R 3.1.3)
##  plyr       * 1.8.1   2014-02-26 CRAN (R 3.1.3)
##  proto      * 0.3-10  2012-12-22 CRAN (R 3.1.3)
##  R6         * 2.0.1   2014-10-29 CRAN (R 3.1.3)
##  Rcpp       * 0.11.5  2015-03-06 CRAN (R 3.1.3)
##  reshape2   * 1.4.1   2014-12-06 CRAN (R 3.1.3)
##  RJSONIO    * 1.3-0   2014-07-28 CRAN (R 3.1.3)
##  rmarkdown  * 0.5.1   2015-01-26 CRAN (R 3.1.3)
##  rstudioapi * 0.2     2014-12-31 CRAN (R 3.1.3)
##  scales     * 0.2.4   2014-04-22 CRAN (R 3.1.3)
##  shiny      * 0.11.1  2015-02-11 CRAN (R 3.1.3)
##  stringr    * 0.6.2   2012-12-06 CRAN (R 3.1.3)
##  xtable     * 1.7-4   2014-09-12 CRAN (R 3.1.3)
##  yaml       * 2.1.13  2014-06-12 CRAN (R 3.1.3)</code></pre>
</div>

<!-- </div>
</div>
</div>
 -->

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
