---
title: Calculating Basic Population Genetic Statistics from SNP Data
author: Stéphanie Manel
date: March 19, 2015
output: html_document
---

Introduction
========================================================

In this vignette, you will  calculate basic population genetic statistics from
SNP data using R packages. This statistics serve as exploratory analysis and
requires to work at population level. We will calculate: 

 i. Genetic diversity,
 ii. Test Hardy Weinberg 
 iii. $F_{is}$ and global $F_{st}$.

The dataset used for those analysis concerns the plant species: lodgepole pine
(*Pinus contorta*, *Pinaceae*). You can have more information on this data set and
the species on the web site of A. Eckert:  (http://eckertdata.blogspot.fr/) But
here the dataset is used as a test dataset with no idea of interpreting the
results in a biological way. We will work on a subset of the dataset to make
the calculations faster.

 

Resources/Packages
========================================================



```{r,packages, message=FALSE}
library(adegenet)
library(hierfstat)
library(pegas)
library(genetics)

```


Workflow for SNP data
========================================================
### Import data

The data are stored in a text file (genotype=AA..). We will  import the dataset
in R as a data frame, and then  convert the SNP data file into a
[genind](http://www.inside-r.org/packages/cran/adegenet/docs/.valid.genind)
object.

The dataset "Master_Pinus_data_genotype.txt" can be downloaded 
[here](https://github.com/NESCent/popgenInfo/tree/master/data/Master_Pinus_data_genotype.txt).

The text file is a matrix of (550 rows x 3086 columns). It contains 4
extra columns: first column is the label of the individuals, the three other are
description of the region, all the other columns are for the genotypes as (AA or
AT...).

When you import the data in R, you need to be in the in the right directory.

```{r, data_import_df_show, eval = FALSE}
Mydata <- read.table("Master_Pinus_data_genotype.txt", header = TRUE)
dim(Mydata) # Matrix of dimension 550x3086
```

```{r, data_import_df_do, echo = FALSE}
Mydata <- read.table("../data/Master_Pinus_data_genotype.txt", header = TRUE)
dim(Mydata) #Matrix of dimension 550x3086
```

To convert RData to a genind object:

We need to create a matrix with genotypes only. We will work on a subset of
data. To convert MydataR to genind object (adegent), we create the object locus
which contains only genotypes We remove the extra columns and we keep only 20
SNP (to make calculation faster). It is also possible to convert Mydata1 into
fstat file (Mydata2), or into loci object (pegas) (Mydata3).

```{r,data_conversion}
locus <- Mydata[, -c(1,2,3,4,16:3086)]    
ind <- as.character(Mydata$tree_id) # individual label
population <- as.character(Mydata$state) # population label
Mydata1 <- df2genind(locus,ploidy=2,ind.names=ind,pop=population)
Mydata1
Mydata1@loc.nall
Mydata2 <- genind2hierfstat(Mydata1) 
Mydata3 <- genind2loci(Mydata1)
```

### Genetic diversity (osberved and expected heterozygosity) 

with adegenet

```{r, Genetic_Diversity}

div <- summary(Mydata1)
div
names(div)

plot(div$Hobs, xlab="Loci number", ylab="Observed Heterozygosity", 
	 main="Observed heterozygosity per locus")

plot(div$Hobs,div$Hexp, xlab="Hobs", ylab="Hexp", 
	 main="Expected heterozygosity in function of observed heterozygosity per locus")

bartlett.test(list(div$Hexp, div$Hobs)) # a test : H0: Hexp = Hobs
```
We observed that heterozygosity varies among loci.
We observed no difference between expected and observed heterozygosity.


Basic statistics with hierfstat. Populations are states. The fuction `basicstat`
provides the observed heterozygosity ($H_o$), mean gene diversities within
population ($H_s$), $F_{is}$, and $F_{st}$ The function `boot.ppfis` provides
confidence interval for $F_{is}$ The fuction `indpca`  does an PCA on the
centered matrix of individual's allele frequencies.

```{r,basic_statistics}
basicstat <- basic.stats(Mydata2, diploid = TRUE, digits = 2) 
names(basicstat)   
boot.ppfis(Mydata2) 
x <- indpca(Mydata2) 
plot(x, cex=0.7)

```

### Testing for Hardy-Weinberg Equilibrium  
We used the pegas package.

```{r,Hardy-Weinberg_test}

hw.test(Mydata3, B = 1000)
```

We get for each locus a test of significance of the null hypothesis: $H_0$ the
locus is in HW equilibrium in the population. All but one locus are in HW
equilibrium.



Conclusion
========================================================
### What did we learn today?

In this vignette, we learned how to explore the patterns of genetic diversity
and how to estimate the level of genetic differentiation in one population.
Also, you have an idea of potential violations of the dataset to the null
Wright-Fischer model.

### What is next?

You may now want to move into looking into population differentiation in more
detail (See [Calculating genetic differentiation and clustering methods from SNP
data](DifferentiationSNP.html))
 
 
Reference
========================================================

Eckert, A. J., A. D. Bower, S. C. González-Martínez, J. L. Wegrzyn, G. Coop and D. B. Neale. 2010. Back to nature: Ecological genomics of loblolly pine (Pinus taeda, Pinaceae). Molecular Ecology 19: 3789-3805.
