---
title: "Calculating Basic Population Genetic Statistics from Sequence Data"
author: "Margarita M. LÃ³pez-Uribe"
date: "March 18, 2015"
output: html_document
---

Introduction
=========
In this vignette, you will see general information on how to calculate basic population genetic statistics from sequence data using R packages. These basic statistics serve as exploratory analyses that will tell you 1) the levels of genetic diversity in your overall population and subpopulations, 2) general patterns of the distribution of genetic diversity across subpopulations, and 3) possible violations of the data to the Wright-Fischer Model (Ref).

The first section of this tutorial provides information on how to analyze sequence data from one locus (link to section 1). The second section will discuss how to analyze multilocus sequence data (link to section 2).


Discuss what kind of questions we are asking
Links to theoretical background
Links to data information
Link to the data to download for the vignette

### Assumptions
  - Sample Size (hudson's Snn)
  - No linkage

Resources/Packages
=========
```{r, packages, message=FALSE}
library(adegenet) # to store the data
library(ape)
library(pegas)
library(hierfstat)
library(mmod)
library(genetics)
```

Section I - Workflow for Raw Sequence Data
========

### Import data
To import raw sequence data into R, the most widely used formats are [FASTA](http://www.ncbi.nlm.nih.gov/BLAST/fasta.html) and nexus. Before you import your data into R, please make sure the sequences are aligned. Sequence alignments can be done in software such as Clustal or Muscle. Both are are part of the software SeaView. Alignments can be saved in FASTA format. (maybe an R option can be library(Biostrings) but I have not used it yet). In all cases, we will convert the sequence data file into [genind](http://www.inside-r.org/packages/cran/adegenet/docs/.valid.genind) objects, which is the standard file format for population genetic analyses in R.



To import raw sequence data into R:
```{r, data_import_FASTA_locus1, eval = FALSE}
myData1 <- read.FASTA("Ebom_mt.fas") # this creates a DNAbin object
class(myData1)
```

```{r, data_import_FASTA_locus1, echo = FALSE}
myData1<- read.FASTA("../data/Ebom_mt.fas") # this creates a DNAbin object
class(myData1)
```

To convert DNAbin objects to genind objects:
```{r, data_import_DNAbin}
#library(mmod)
tmp1<-as.matrix(myData1) # converts the list of sequences into a matrix
myData1b<-as.genind.DNAbin(tmp1, rep(c("CA","Ch","Am","AF"), each=5))
```

### Genetic diversity indices
```{r, pi}
myData1_pi<-nuc.div(myData1) # nucleotide diversity from a sample of DNA sequences in DNAbin format
myData1_pi
myData1_summ<-summary(myData1b) # to visualize summary descriptors of your sequence data file
par(mfrow=c(2,2))
plot(myData1_summ$pop.eff, myData1_summ$pop.nall, xlab="Sample size",
ylab="Number of alleles",main="Alleles numbers and sample sizes",
type="n")
text(myData1_summ$pop.eff,myData1_summ$pop.nall,lab=names(myData1_summ$pop.eff))
barplot(myData1_summ$loc.nall, ylab="Number of alleles",
main="Number of alleles per locus")
barplot(myData1_summ$Hexp-myData1_summ$Hobs, main="Heterozygosity: expected-observed",
ylab="Hexp - Hobs")
barplot(myData1_summ$pop.eff, main="Sample sizes per population",
ylab="Number of genotypes",las=3)
```

Note that no observed heterozygosity is expected because sequences come from mitochondrial data.

### Testing for Hardy-Weinberg Equilibrium
```{r, HWE1, message=FALSE, warning=FALSE}
#library(genetics)
myData1_HWE<- HWE.test.genind(myData1b) # this will test for Hardy-Weingber equilibrium
myData1_HWE # p-values lower than 0.05 indicate deviations from HWE
```
Note: If you have loci in your dataset that deviate from HWE, your dataset may violate assumptions for downstream population genetic analyses.

### F statistics
```{r, Fstats}
myData1_loci<- as.loci(myData1b) 
Fst(myData1_loci) # this function calculates overall Fit, Fst, and Fis of the dataset
```

### Testing for Population Differentiation

#### Testing for Population Differentiation from diploid data
```{r, Fst1}
myData1_bs<- chao_bootstrap(myData1b, nreps=100) # subsample your dataset multiple times to estimate significance of genetic differentation statistics 
summarise_bootstrap(myData1_bs, Gst_Nei) # estimate Gst_Nei and whether or not it is significant
summarise_bootstrap(myData1_bs, Gst_Hedrick) # estimate Hedrick's Gst and whether or not it is significant
summarise_bootstrap(myData1_bs, D_Jost) # estimate Jost's D and whether or not it is significant
myData1_GenDiff<-diff_stats(myData1b) # general test for genetic differentiation per locus and across all subpopultions using Gst, Gst' and Jost D (estimated and based on the harmonic mean between loci)
myData1_GenDiff
```

#### Testing for Population Differentiation from haploid data
```{r, Fst2}
#library(mmod)
myData1_Phist<- Phi_st_Meirmans(myData1b) 
myData1_Phist # test fo population differentation for haploid data
```



Section II - Workflow for Multilocus Sequence Data 
========
### Import data

To import raw sequence data from more loci into R:
```{r, data_import_FASTA_locus2}
myData2<- read.FASTA("Ebom_CAD.fas") # this creates a DNAbin object
class(myData2)
tmp2<-as.matrix(myData2) # converts the list of sequences into a matrix
allData<-cbind(tmp1,tmp2)
```

To convert DNAbin objects to genind objects:
```{r, data_import_DNAbin_all}
#library(mmod)
tmp3<-as.matrix(allData) # converts the list of sequences into a matrix
allDatab<-as.genind.DNAbin(tmp3, rep(c("CA","Ch","Am","AF"), each=5)) #bins data from multiple loci into one DNAbin object - MAKE SURE SEQUENCES IN THE FASTA FILE HAVE THE SAME NAME
```

#Analyses
For example, testing for Genetic Differentiation:
```{r, HWE2, message=FALSE}
#library(genetics)
allData_HWE<- HWE.test.genind(allDatab, res="matrix") # this will generate a matrix of p-values for HWE
allData_HWE

```

### F statistics
```{r, Fstats2}
allData_loci<- as.loci(allDatab) 
Fst(allData_loci) # this function calculates overall Fit, Fst, and Fis of the dataset
```

Conclusions
========

### What did we learn today?
In this vignette, we learned how to explore the patterns of genetic diveristy in ou dataset. Now, you know the level of genetic differentiation in your population and subpopulations, and whether or now they are genetically different. Also, you have an idea of potentiall violations of the dataset to the null Wright-Fischer model. 

### What is next?
You may now want to move into looking into population differentation in more detail (link to vignette). You may also want to explore demographic parameter estimation (e.g. deviations from neutrality,effective population size).
